<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tree Thinking • apb2019</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Tree Thinking">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">apb2019</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/RB_Discrete_Morphology/RB_MCMC_Discrete_Morph.html">Discrete morphology - Tree Inference</a>
    </li>
    <li>
      <a href="../../articles/RB_Total_Evidence/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/00-installs.html">Installs</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/03-FossilSim.html">FossilSim FBD Exercise</a>
    </li>
    <li>
      <a href="../../articles/lectures/BayesianModels.html">Bayesian Models</a>
    </li>
    <li>
      <a href="../../articles/lectures/CompMethods.html">Extended Morphology Models</a>
    </li>
    <li>
      <a href="../../articles/lectures/TreeThinking.html">Tree Thinking</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Tree Thinking</h1>
                        <h4 class="author">April Wright</h4>
            
            <h4 class="date">7.17.19</h4>
      
      
      <div class="hidden name"><code>TreeThinking.Rmd</code></div>

    </div>

    
    
<div id="good-morning" class="section level2">
<h2 class="hasAnchor">
<a href="#good-morning" class="anchor"></a>Good Morning!</h2>
<ul>
<li>What is a tree?</li>
<li>How is a tree built?</li>
<li>What are phylogenetic data?</li>
</ul>
</div>
<div id="what-do-we-do-with-a-phylogeny" class="section level2">
<h2 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny" class="anchor"></a>What do we do with a phylogeny?</h2>
<ul>
<li>Determine the timing of trait evolution</li>
</ul>
<p><img src="img/skinks.png" align="middle" height="350" width="350" margin="0 auto"></p>
<p>Skink tree from Wright et al. 2015</p>
</div>
<div id="what-do-we-do-with-a-phylogeny-1" class="section level2">
<h2 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-1" class="anchor"></a>What do we do with a phylogeny?</h2>
<ul>
<li>Tell homology from convergence</li>
</ul>
<p><img src="img/A_spinner_dolphin_in_the_Red_Sea.jpg" align="left" height="350" width="350" margin="0 auto"><img src="img/skeleton.jpg" align="right" height="250" width="350" margin="0 auto"> Dolphin, Alex Vasenin via WikiMedia</p>
</div>
<div id="what-do-we-do-with-a-phylogeny-2" class="section level2">
<h2 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-2" class="anchor"></a>What do we do with a phylogeny?</h2>
<ul>
<li>Trace the origins of structures <img src="img/AskABiologist.png" align="right" height="750" width="350" margin="0 auto">
</li>
</ul>
<p>Ask a Biologist illustration of homology</p>
</div>
<div id="what-do-we-do-with-a-phylogeny-3" class="section level2">
<h2 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-3" class="anchor"></a>What do we do with a phylogeny?</h2>
<ul>
<li><p>Taxonomy</p></li>
<li>
<p>Hennig, 1950 Grundzüge einer Theorie der Phylogenetischen Systematik</p>
<ul>
<li>Taxonomy should be logically consistent with the tree for the group</li>
</ul>
</li>
</ul>
</div>
<div id="what-do-we-do-with-a-phylogeny-4" class="section level2">
<h2 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-4" class="anchor"></a>What do we do with a phylogeny?</h2>
<ul>
<li><p>Taxonomy</p></li>
<li>
<p>Hennig, 1950 Grundzüge einer Theorie der Phylogenetischen Systematik</p>
<ul>
<li>Taxonomy should be logically consistent with the tree for the group</li>
</ul>
</li>
<li>
<p>Sneath &amp; Sokal, 1963, 1973</p>
<ul>
<li>Using distance matrices to cluster based on phenetic similarity</li>
</ul>
</li>
</ul>
</div>
<div id="tree-terms-tip" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-tip" class="anchor"></a>Tree Terms: Tip</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phytools)</span></code></pre></div>
<pre><code>## Loading required package: ape</code></pre>
<pre><code>## Loading required package: maps</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>tree &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/phytools/topics/pbtree">pbtree</a></span>(<span class="dt">n =</span> <span class="dv">5</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<p>Tip: What we are putting on the tree. May be species, individuals, or higher-order taxa. May be called terminal node, leaf, one degree node.</p>
</div>
<div id="tree-terms-tip-1" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-tip-1" class="anchor"></a>Tree Terms: Tip</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>tree<span class="op">$</span>tip.label</span></code></pre></div>
<pre><code>## [1] "t2" "t3" "t4" "t5" "t1"</code></pre>
<pre><code>[1] "t4" "t5" "t1" "t2" "t3"</code></pre>
<p>Access in R: tree$tip.label</p>
</div>
<div id="tree-terms-node" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-node" class="anchor"></a>Tree Terms: Node</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">nodelabels</span>(<span class="dt">cex=</span><span class="dv">4</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>Node: Where nodes meet, implying a most recent common ancestor. May be called vertex, or three-degree node.</p>
</div>
<div id="tree-terms-node-1" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-node-1" class="anchor"></a>Tree Terms: Node</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ape)</span>
<span id="cb10-2"><a href="#cb10-2"></a>tree<span class="op">$</span>Nnode</span></code></pre></div>
<pre><code>## [1] 4</code></pre>
<pre><code>[1] 4</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/ape/topics/mrca">getMRCA</a></span>(tree, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"t1"</span>, <span class="st">"t2"</span>))</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<pre><code>[1] 6</code></pre>
</div>
<div id="tree-terms-branch" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-branch" class="anchor"></a>Tree Terms: branch</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co">#plot(tree, cex = 3.5, no.margin = TRUE, edge.width = 1.5)</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>tree<span class="op">$</span>edge</span></code></pre></div>
<pre><code>     [,1] [,2]
[1,]    6    7
[2,]    7    8
[3,]    8    1
[4,]    8    2
[5,]    7    3
[6,]    6    9
[7,]    9    4
[8,]    9    5</code></pre>
<p>Branch: What connects the tip to the tree. Can have a variety of units, which we will discuss over the next few days. May be called edge. Access in R: tree$edge</p>
</div>
<div id="tree-terms-branch-1" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-branch-1" class="anchor"></a>Tree Terms: branch</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phytools)</span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/phytools/topics/plotBranchbyTrait">plotBranchbyTrait</a></span>(tree,tree<span class="op">$</span>edge.length,<span class="dt">method=</span><span class="st">"tips"</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>tree<span class="op">$</span>edge.length</span></code></pre></div>
<pre><code>## [1] 1.2144820 0.7489669 0.4599640 0.4599640 1.0385148 0.1704161 0.1704161
## [8] 2.4234130</code></pre>
<pre><code>[1] 0.01070042 1.00622147 0.36177554 0.36177554 1.36799701 0.23368315
[7] 1.14501427 1.14501427</code></pre>
</div>
<div id="tree-terms" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms" class="anchor"></a>Tree Terms</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>, <span class="dt">direction =</span> <span class="st">"downwards"</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="tree-terms-1" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-1" class="anchor"></a>Tree Terms</h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>, <span class="dt">type=</span><span class="st">"fan"</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div id="tree-terms-rotation---reflecting-taxa-at-a-node" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-rotation---reflecting-taxa-at-a-node" class="anchor"></a>Tree Terms: Rotation - reflecting taxa at a node</h2>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/ape/topics/nodelabels">nodelabels</a></span>(<span class="dt">cex =</span> <span class="fl">3.5</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/phytools/topics/rotateNodes">rotateNodes</a></span>(tree, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">7</span>, <span class="dv">8</span>))</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 5 tips and 4 internal nodes.
## 
## Tip labels:
## [1] "t4" "t5" "t3" "t2" "t1"
## 
## Rooted; includes branch lengths.</code></pre>
<pre><code>
Phylogenetic tree with 5 tips and 4 internal nodes.

Tip labels:
[1] "t1" "t5" "t4" "t2" "t3"

Rooted; includes branch lengths.</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="tree-terms-monophyletic---an-ancestor-and-all-its-descendents" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-monophyletic---an-ancestor-and-all-its-descendents" class="anchor"></a>Tree Terms: Monophyletic - an ancestor and all its descendents</h2>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/ape/topics/is.monophyletic">is.monophyletic</a></span>(tree, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"t1"</span>, <span class="st">"t2"</span>), <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<pre><code>## [1] FALSE</code></pre>
<pre><code>[1] FALSE</code></pre>
</div>
<div id="tree-terms-rooting" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-rooting" class="anchor"></a>Tree Terms: Rooting</h2>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="co"># reroot(tree, node.number)</span></span>
<span id="cb32-2"><a href="#cb32-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>Ingroup: Taxa of interest</p>
<p>Outgroup: Taxon closely related used to root the tree</p>
</div>
<div id="tree-terms-rooting-1" class="section level2">
<h2 class="hasAnchor">
<a href="#tree-terms-rooting-1" class="anchor"></a>Tree Terms: Rooting</h2>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>unroot_tree &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/ape/topics/root">unroot</a></span>(tree)</span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(unroot_tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div id="how-is-a-tree-built" class="section level2">
<h2 class="hasAnchor">
<a href="#how-is-a-tree-built" class="anchor"></a>How is a tree built?</h2>
<ul>
<li>Many ways. We will focus on three:
<ul>
<li>Maximum parsimony</li>
<li>Maximum likelihood</li>
<li>Bayesian inference</li>
</ul>
</li>
</ul>
</div>
<div id="phylogenetic-data" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data" class="anchor"></a>Phylogenetic Data</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(alignfigR)</span></code></pre></div>
<pre><code>## Welcome to alignfigR!</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>char_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(<span class="st">"../../data/bears_fasta.fa"</span>)</span>
<span id="cb36-2"><a href="#cb36-2"></a>char_data[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## $Agriarctos_spp
##  [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
## [18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
## [35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
## [52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## 
## $Ailurarctos_lufengensis
##  [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## [18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
## [35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
## [52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## 
## $Ailuropoda_melanoleuca
##  [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
## [18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
## [35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
## [52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
</div>
<div id="phylogenetic-data-1" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-1" class="anchor"></a>Phylogenetic Data</h2>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(alignfigR)</span>
<span id="cb39-2"><a href="#cb39-2"></a>char_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(<span class="st">"../../data/bears_fasta.fa"</span>)</span>
<span id="cb39-3"><a href="#cb39-3"></a>char_data[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## $Agriarctos_spp
##  [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
## [18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
## [35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
## [52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## 
## $Ailurarctos_lufengensis
##  [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## [18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
## [35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
## [52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## 
## $Ailuropoda_melanoleuca
##  [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
## [18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
## [35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
## [52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<p>These data are binary</p>
</div>
<div id="phylogenetic-data-2" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-2" class="anchor"></a>Phylogenetic Data</h2>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(alignfigR)</span>
<span id="cb42-2"><a href="#cb42-2"></a>char_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(<span class="st">"../../data/bears_fasta.fa"</span>)</span>
<span id="cb42-3"><a href="#cb42-3"></a>char_data[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## $Agriarctos_spp
##  [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
## [18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
## [35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
## [52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## 
## $Ailurarctos_lufengensis
##  [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## [18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
## [35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
## [52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"
## 
## $Ailuropoda_melanoleuca
##  [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
## [18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
## [35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
## [52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<p>Always arranged with rows being taxa and columns corresponding to a character - “matrix” structure</p>
</div>
<div id="phylogenetic-data-3" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-3" class="anchor"></a>Phylogenetic Data</h2>
<p>Text editor - phylo data, metadata</p>
</div>
<div id="phylogenetic-data-4" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-4" class="anchor"></a>Phylogenetic Data</h2>
</div>
<div id="phylogenetic-data-5" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-5" class="anchor"></a>Phylogenetic Data</h2>
<p>DNA data tends to be simple</p>
</div>
<div id="phylogenetic-data-6" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-6" class="anchor"></a>Phylogenetic Data</h2>
<p>Example character from Brady:</p>
<ol start="3" style="list-style-type: decimal">
<li>Worker, queen, and male. Specialized, stout setae on anterior margin of clypeus: (0) absent; (1) present. The presence of these specialized setae is a putative synapomorphy of Amblyoponinae (Ward, 1994), including Amblyopone and Onychomyrmex. <img src="img/2000px-Front_of_insect_head_diagram.svg.png" align="right" height="250" width="350" margin="0 auto">
</li>
</ol>
<p>image via Ask a Biologist</p>
</div>
<div id="phylogenetic-data-7" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-7" class="anchor"></a>Phylogenetic Data</h2>
<ul>
<li>How do we know we have a truly discrete state? <img src="img/2000px-Front_of_insect_head_diagram.svg.png" align="right" height="250" width="350" margin="0 auto">
</li>
</ul>
</div>
<div id="phylogenetic-data-8" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-8" class="anchor"></a>Phylogenetic Data</h2>
<p>How do we know we’ve captured the relevant character axes? <img src="img/butterfly_human_vision.jpg" align="right" height="250" width="350" margin="0 auto"></p>
<p>image via Ask a biologist, Mike Hagelberg</p>
</div>
<div id="phylogenetic-data-9" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-9" class="anchor"></a>Phylogenetic Data</h2>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</span>
<span id="cb45-2"><a href="#cb45-2"></a>colors &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"blue"</span>, <span class="st">"purple"</span>,<span class="st">"white"</span>)</span>
<span id="cb45-3"><a href="#cb45-3"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/alignfigR/topics/plot_alignment">plot_alignment</a></span>(char_data, colors, <span class="dt">taxon_labels =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">text =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="dt">size=</span><span class="dv">40</span>))</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
<div id="phylogenetic-data-10" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-data-10" class="anchor"></a>Phylogenetic Data</h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</span>
<span id="cb46-2"><a href="#cb46-2"></a>colors &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"blue"</span>, <span class="st">"purple"</span>,<span class="st">"white"</span>)</span>
<span id="cb46-3"><a href="#cb46-3"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/alignfigR/topics/plot_alignment">plot_alignment</a></span>(char_data, colors, <span class="dt">taxon_labels =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">text =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="dt">size=</span><span class="dv">40</span>))</span></code></pre></div>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
<p>How do we go from this to a tree?</p>
</div>
<div id="parsimony" class="section level2">
<h2 class="hasAnchor">
<a href="#parsimony" class="anchor"></a>Parsimony</h2>
<ul>
<li>Not only applied in phylogenetics</li>
<li>The simplest explanation for the observed data is the best</li>
</ul>
</div>
<div id="parsimony-1" class="section level2">
<h2 class="hasAnchor">
<a href="#parsimony-1" class="anchor"></a>Parsimony</h2>
<ul>
<li>Maximum parsimony: the tree that minimizes the number of “steps”, or changes, on a tree is to be preferred</li>
<li>Let’s turn to the board for a minute: Parsimony informative, invariant, and parsimony non-informative variation</li>
</ul>
<p>??? Have them start installs on the next page while we do this.</p>
</div>
<div id="treesiftr" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr" class="anchor"></a>treesiftr</h2>
<p>RStudio –or–Shiny</p>
</div>
<div id="treesiftr-1" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr-1" class="anchor"></a>treesiftr</h2>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(treesiftr)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by 'treeio':
##   method     from
##   root.phylo ape</code></pre>
<pre><code>## 
## Attaching package: 'treesiftr'</code></pre>
<pre><code>## The following object is masked _by_ '.GlobalEnv':
## 
##     tree</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>aln_path &lt;-<span class="st"> "../../data/bears_fasta.fa"</span></span>
<span id="cb51-2"><a href="#cb51-2"></a>bears &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(aln_path)</span>
<span id="cb51-3"><a href="#cb51-3"></a>bear_tree &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/ape/topics/multi2di">multi2di</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/ape/topics/read.tree">read.tree</a></span>(<span class="st">"../../data/starting_tree.tre"</span>))</span>
<span id="cb51-4"><a href="#cb51-4"></a>sample_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/treesiftr/topics/generate_sliding">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">1</span>, <span class="dt">stop_char =</span> <span class="dv">5</span>, <span class="dt">steps =</span> <span class="dv">1</span>)</span>
<span id="cb51-5"><a href="#cb51-5"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(sample_df)</span></code></pre></div>
<pre><code>##   starting_val stop_val step_val
## 1            1        2        1
## 2            2        3        1
## 3            3        4        1
## 4            4        5        1
## 5            5        6        1</code></pre>
<pre><code>  starting_val stop_val step_val
1            1        2        1
2            2        3        1
3            3        4        1
4            4        5        1
5            5        6        1</code></pre>
</div>
<div id="treesiftr-2" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr-2" class="anchor"></a>treesiftr</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phangorn)</span>
<span id="cb54-2"><a href="#cb54-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggtree)</span></code></pre></div>
<pre><code>## ggtree v1.16.3  For help: https://yulab-smu.github.io/treedata-book/
## 
## If you use ggtree in published research, please cite the most appropriate paper(s):
## 
## [36m-[39m Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, accepted. doi: 10.1093/molbev/msy194[36m-[39m Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36, doi:10.1111/2041-210X.12628</code></pre>
<pre><code>## 
## Attaching package: 'ggtree'</code></pre>
<pre><code>## The following object is masked from 'package:phytools':
## 
##     read.newick</code></pre>
<pre><code>## The following object is masked from 'package:ape':
## 
##     rotate</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a>output_vector &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/treesiftr/topics/generate_tree_vis">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df,                                        <span class="dt">alignment =</span> aln_path, </span>
<span id="cb59-2"><a href="#cb59-2"></a>                                   <span class="dt">tree =</span> bear_tree, </span>
<span id="cb59-3"><a href="#cb59-3"></a>                                   <span class="dt">phy_mat =</span> bears,                                              <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## Generating tree for charset:12</code></pre>
<pre><code>## Final p-score 2 after  0 nni operations</code></pre>
<pre><code>## Generating tree for charset:23</code></pre>
<pre><code>## Final p-score 2 after  0 nni operations</code></pre>
<pre><code>## Generating tree for charset:34</code></pre>
<pre><code>## Final p-score 2 after  0 nni operations</code></pre>
<pre><code>## Generating tree for charset:45</code></pre>
<pre><code>## Final p-score 2 after  0 nni operations</code></pre>
<pre><code>## Generating tree for charset:56</code></pre>
<pre><code>## Final p-score 2 after  0 nni operations</code></pre>
</div>
<div id="treesiftr-3" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr-3" class="anchor"></a>treesiftr</h2>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1"></a>output_vector <span class="co">#sample output - you will get more than this when you run in your console</span></span></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]
## NULL
## 
## [[5]]</code></pre>
<p><img src="TreeThinking_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
</div>
<div id="parsimony-many-trees-for-one-character-and-4-taxa" class="section level2">
<h2 class="hasAnchor">
<a href="#parsimony-many-trees-for-one-character-and-4-taxa" class="anchor"></a>Parsimony: Many trees for one character and 4 taxa</h2>
<p><img src="img/Ptree_enum.png" align="right" height="350" width="500" margin="0 auto"></p>
<p>Parsimony Trees</p>
<p>This is one character. Imagine many - enumeration is not possible.Also note that several trees have the same “best” tree</p>
<p>Image via Mark Holder</p>
</div>
<div id="parsimony-how-do-we-find-the-most-parsimonious-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#parsimony-how-do-we-find-the-most-parsimonious-tree" class="anchor"></a>Parsimony: How do we find the most parsimonious tree?</h2>
<ul>
<li>We’re going to take an exercise break and play with PAUP
<ul>
<li><a href="http://phylosolutions.com/paup-test/" class="uri">http://phylosolutions.com/paup-test/</a></li>
<li>Download the command line executeable for your OS</li>
<li>Copy it into this repository</li>
<li>Open a terminal</li>
</ul>
</li>
</ul>
</div>
<div id="paup" class="section level2">
<h2 class="hasAnchor">
<a href="#paup" class="anchor"></a>PAUP</h2>
<pre class="paup"><code>execute data/bears_morphology.nex</code></pre>
<ul>
<li>NOTE: PAUP allows tab-completion</li>
<li>Open the bears_morphology file in a text editor. Now:</li>
</ul>
</div>
<div id="paup-a-couple-important-commands" class="section level2">
<h2 class="hasAnchor">
<a href="#paup-a-couple-important-commands" class="anchor"></a>PAUP: A couple important commands</h2>
<pre class="paup"><code>cstatus
tstatus
showmatrix
showdist
log file="mylogfile"</code></pre>
<ul>
<li>Try each of these - what information do they give you?</li>
</ul>
</div>
<div id="paup-building-a-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#paup-building-a-tree" class="anchor"></a>PAUP: Building a tree</h2>
<pre class="paup"><code>alltrees</code></pre>
<p>What happened here?</p>
</div>
<div id="parsimony-enumeration-is-not-possible-for-more-than-12-taxa" class="section level2">
<h2 class="hasAnchor">
<a href="#parsimony-enumeration-is-not-possible-for-more-than-12-taxa" class="anchor"></a>Parsimony: Enumeration is not possible for more than 12 taxa</h2>
<p><img src="img/Ptree_enum.png" align="center" height="500" width="750" margin="0 auto"></p>
<p>Parsimony Trees</p>
</div>
<div id="paup-heuristic-searches" class="section level2">
<h2 class="hasAnchor">
<a href="#paup-heuristic-searches" class="anchor"></a>PAUP: Heuristic Searches</h2>
<p>Heuristic - use of shortcuts to reduce the number of trees we need to search</p>
<pre class="paup"><code>hsearch</code></pre>
<ul>
<li>What is the name of the heuristic that was used?</li>
<li>How was the initial tree discovered?</li>
<li>How many trees were searched?</li>
<li>How many “best” trees were there, and what is their score?</li>
</ul>
</div>
<div id="paup-heuristic-searches-1" class="section level2">
<h2 class="hasAnchor">
<a href="#paup-heuristic-searches-1" class="anchor"></a>PAUP: Heuristic Searches</h2>
<p>Heuristic - use of shortcuts to reduce the number of trees we need to search</p>
<pre class="paup"><code>hsearch swap = nni</code></pre>
<ul>
<li>How many trees were examined with this algorithm? Why is this number so much smaller?</li>
<li>How many “best” trees were found, and what is their score?</li>
</ul>
</div>
<div id="paup-heuristic-searches-2" class="section level2">
<h2 class="hasAnchor">
<a href="#paup-heuristic-searches-2" class="anchor"></a>PAUP: Heuristic Searches</h2>
<p>Heuristic - use of shortcuts to reduce the number of trees we need to search</p>
<pre class="paup"><code>hsearch swap = spr</code></pre>
<ul>
<li>How many trees were examined with this algorithm?</li>
<li>How many “best” trees were found, and what is their score?</li>
<li>When would we expect searching algorithm to matter strongly?</li>
</ul>
</div>
<div id="paup-exporting-parsimony-trees" class="section level2">
<h2 class="hasAnchor">
<a href="#paup-exporting-parsimony-trees" class="anchor"></a>PAUP: Exporting parsimony trees</h2>
<pre class="paup"><code>savetrees from=1 to=1 file=results/tree1.tre;
savetrees from=2 to=2 file=results/tree2.tre;
savetrees from=3 to=3 file=results/tree3.tre;
</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#good-morning">Good Morning!</a></li>
      <li><a href="#what-do-we-do-with-a-phylogeny">What do we do with a phylogeny?</a></li>
      <li><a href="#what-do-we-do-with-a-phylogeny-1">What do we do with a phylogeny?</a></li>
      <li><a href="#what-do-we-do-with-a-phylogeny-2">What do we do with a phylogeny?</a></li>
      <li><a href="#what-do-we-do-with-a-phylogeny-3">What do we do with a phylogeny?</a></li>
      <li><a href="#what-do-we-do-with-a-phylogeny-4">What do we do with a phylogeny?</a></li>
      <li><a href="#tree-terms-tip">Tree Terms: Tip</a></li>
      <li><a href="#tree-terms-tip-1">Tree Terms: Tip</a></li>
      <li><a href="#tree-terms-node">Tree Terms: Node</a></li>
      <li><a href="#tree-terms-node-1">Tree Terms: Node</a></li>
      <li><a href="#tree-terms-branch">Tree Terms: branch</a></li>
      <li><a href="#tree-terms-branch-1">Tree Terms: branch</a></li>
      <li><a href="#tree-terms">Tree Terms</a></li>
      <li><a href="#tree-terms-1">Tree Terms</a></li>
      <li><a href="#tree-terms-rotation---reflecting-taxa-at-a-node">Tree Terms: Rotation - reflecting taxa at a node</a></li>
      <li><a href="#tree-terms-monophyletic---an-ancestor-and-all-its-descendents">Tree Terms: Monophyletic - an ancestor and all its descendents</a></li>
      <li><a href="#tree-terms-rooting">Tree Terms: Rooting</a></li>
      <li><a href="#tree-terms-rooting-1">Tree Terms: Rooting</a></li>
      <li><a href="#how-is-a-tree-built">How is a tree built?</a></li>
      <li><a href="#phylogenetic-data">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-1">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-2">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-3">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-4">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-5">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-6">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-7">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-8">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-9">Phylogenetic Data</a></li>
      <li><a href="#phylogenetic-data-10">Phylogenetic Data</a></li>
      <li><a href="#parsimony">Parsimony</a></li>
      <li><a href="#parsimony-1">Parsimony</a></li>
      <li><a href="#treesiftr">treesiftr</a></li>
      <li><a href="#treesiftr-1">treesiftr</a></li>
      <li><a href="#treesiftr-2">treesiftr</a></li>
      <li><a href="#treesiftr-3">treesiftr</a></li>
      <li><a href="#parsimony-many-trees-for-one-character-and-4-taxa">Parsimony: Many trees for one character and 4 taxa</a></li>
      <li><a href="#parsimony-how-do-we-find-the-most-parsimonious-tree">Parsimony: How do we find the most parsimonious tree?</a></li>
      <li><a href="#paup">PAUP</a></li>
      <li><a href="#paup-a-couple-important-commands">PAUP: A couple important commands</a></li>
      <li><a href="#paup-building-a-tree">PAUP: Building a tree</a></li>
      <li><a href="#parsimony-enumeration-is-not-possible-for-more-than-12-taxa">Parsimony: Enumeration is not possible for more than 12 taxa</a></li>
      <li><a href="#paup-heuristic-searches">PAUP: Heuristic Searches</a></li>
      <li><a href="#paup-heuristic-searches-1">PAUP: Heuristic Searches</a></li>
      <li><a href="#paup-heuristic-searches-2">PAUP: Heuristic Searches</a></li>
      <li><a href="#paup-exporting-parsimony-trees">PAUP: Exporting parsimony trees</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
