<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Discrete morphology - Relaxed Mk • apb2019</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Discrete morphology - Relaxed Mk">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">apb2019</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/RB_Discrete_Morphology/RB_MCMC_Discrete_Morph.html">Discrete morphology - Tree Inference</a>
    </li>
    <li>
      <a href="../../articles/RB_Extended_morph/RB_MCMC_Extended_Morph.html">Discrete morphology - Relaxed Mk</a>
    </li>
    <li>
      <a href="../../articles/RB_Total_Evidence/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/00-installs.html">Installs</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/03-FossilSim.html">FossilSim FBD Exercise</a>
    </li>
    <li>
      <a href="../../articles/lectures/BayesianModels.html">Bayesian Models</a>
    </li>
    <li>
      <a href="../../articles/lectures/CompMethods.html">Extended Morphology Models</a>
    </li>
    <li>
      <a href="../../articles/lectures/ExtendedMorph.html">Extended Morphology Models</a>
    </li>
    <li>
      <a href="../../articles/lectures/TreeThinking.html">Tree Thinking</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Discrete morphology - Relaxed Mk</h1>
            <h3 class="subtitle">Phylogenetic inference with discrete morphological data</h3>
            
      
      
      <div class="hidden name"><code>RB_MCMC_Extended_Morph.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>While molecular data have become the default for building phylogenetic trees for many types of evolutionary analysis, morphological data remains important, particularly for analyses involving fossils. The use of morphological data raises special considerations for model-based methods for phylogenetic inference. Morphological data are typically collected to maximize the number of parsimony-informative characters - that is, the characters that may provide information in favor of one topology over another. Morphological characters also do not carry common meanings from one character in a matrix to the next; character codings are made arbitrarily. These two factors require extensions to our existing phylogenetic models. Accounting for the complexity of morphological characters remains challenging. This tutorial will provide a discussion of modeling morphological characters, and will demonstrate how to perform Bayesian phylogenetic analysis with morphology using RevBayes.</p>
<div id="graphical-models" class="section level2">
<h2 class="hasAnchor">
<a href="#graphical-models" class="anchor"></a>Graphical Models</h2>
<p><a href="http://revbayes.com">RevBayes</a> uses a <em>graphical model</em> framework in which all probabilistic models, including phylogenetic models, are comprised of modular components that can be assembled in a myriad of ways.</p>
<p>The statistics literature has developed a rich visual representation for graphical models. Visually representing graphical models can be useful for communication model assumptions.</p>
<p>Representing graphical models in computer code (using the <code>Rev</code> language) is useful in developing an understanding of phylogenetic models. For more information about this topic see the tutorial <a href="https://revbayes.github.io/tutorials/intro_graph_models/">Introduction to Graphical Models</a>.</p>
<p>{% figure legend %} <img src="img/graphical_model_legend.png" width="400"><br>
{% figcaption %} *The symbols for a visual representation of a graphical model. a) Solid squares represent constant nodes, which specify fixed-valued variables. b) Stochastic nodes are represented by solid circles. These variables correspond to random variables and may depend on other variables. c) Deterministic nodes (dotted circles) indicate variables that are determined by a specific function applied to another variable. They can be thought of as variable transformations. d) Observed states are placed in clamped stochastic nodes, represented by gray-shaded circles. e) Replication over a set of variables is indicated by enclosing the replicated nodes in a plate (dashed rectangle). f) Tree plates represent the different classes of nodes in a phylogeny. The tree topology orders the nodes in the tree plate and may be a constant node (as in this example) or a stochastic node (if the topology node is a solid circle).</p>
</div>
<div id="overview-of-discrete-morphology-models" class="section level2">
<h2 class="hasAnchor">
<a href="#overview-of-discrete-morphology-models" class="anchor"></a>Overview of Discrete Morphology Models</h2>
<p>{% figure mk_graphical_model %} <img src="img/tikz/Mk_model.png"> {% figcaption %} Graphical model showing the Mk model (left panel). Rev code specifying the Mk model is on the right-hand panel. {% endfigcaption %} {% endfigure %}</p>
<p>Molecular data forms the basis of most phylogenetic analyses today. However, morphological characters remain relevant. Fossils often provide our only direct observation of extinct biodiversity. DNA degradation can make it difficult or impossible to obtain sufficient molecular data from fragile museum specimens. Using morphological data can help researchers include specimens in their phylogeny that might be left out of a molecular tree.</p>
<p>To understand how morphological characters are modeled, it is important to understand how characters are collected. Unlike in molecular data, for which homology is algorithmically determined, homology in a morphological character is typically assessed by an expert. Biologists will typically decide what characters are homologous by looking across specimens at the same structure in multiple taxa; they may also look at the developmental origin of structures in making this assessment. Once homology is determined, characters are broken down into states, or different forms a single character can take. The state <code>0</code> commonly refers to absence, meaning that character is not present. In some codings, absence will mean that character has not evolved in that group. In others, absence means that that character has not evolved in that group, and/or that that character has been lost in that group. This type of coding is arbitrary, but both <em>non-random</em> and <em>meaningful</em>, and poses challenges for how we model the data.</p>
<p>Historically, most phylogenetic analyses using morphological characters have been performed using the maximum parsimony optimality criterion. Maximum parsimony analysis involves proposing trees from the morphological data. Each tree is evaluated according to how many changes it implied in the data, and the tree that requires the fewest changes is preferred. In this way of estimating a tree, a character that does not change, or changes only in one taxon, cannot be used to discriminate between trees (i.e., it does not favor a topology). Therefore, workers with parsimony typically do not collect characters that are parsimony uninformative.</p>
<p>In 2001, Paul Lewis introduced a generalization of the Jukes-Cantor model of sequence evolution for use with morphological data. This model, called the Mk (Markov model, assuming each character is in one of <em>k</em> states) model provided a mathematical formulation that could be used to estimate trees from morphological data in both likelihood and Bayesian frameworks. While this model is a useful step forward, as a generalization of the Jukes-Cantor, it still makes fairly simplistic assumptions. This tutorial will guide you through estimating a phylogeny with the Mk model, and two useful extensions to the model.</p>
</div>
<div id="the-mk-model" class="section level2">
<h2 class="hasAnchor">
<a href="#the-mk-model" class="anchor"></a>The Mk Model</h2>
<p>The Mk model is a generalization of the Jukes-Cantor model of nucleotide sequence evolution.</p>
<p>The Q-matrix for a two-state Mk model looks like so:</p>
<p><span class="math display">\[Q = \begin{pmatrix} -\mu_0 &amp; \mu_{01} \\
\mu_{10} &amp; -\mu_1  &amp;\\
\end{pmatrix} \mbox{  ,}\]</span></p>
<p>This matrix can be expanded to accommodate multi-state data, as well:</p>
<p><span class="math display">\[Q = \begin{pmatrix} -\mu_0 &amp; \mu_{01} &amp; \mu_{02} &amp; \mu_{03} \\
\mu_{10} &amp; -\mu_1  &amp; \mu_{12} &amp; \mu_{13} \\
\mu_{20} &amp; \mu_{21} &amp; -\mu_2  &amp; \mu_{23} \\
\mu_{30} &amp; \mu_{31} &amp; \mu_{32} &amp; -\mu_3 
\end{pmatrix} \mbox{  ,}\]</span></p>
<p>However, the Mk model sets transitions to be equal from any state to any other state. In that sense, our multistate matrix really looks like this:</p>
<p><span class="math display">\[Q = \begin{pmatrix} -(k-1)\mu &amp; \mu &amp; \mu &amp; \mu \\
\mu &amp; -(k-1)\mu  &amp; \mu &amp; \mu \\
\mu &amp; \mu &amp; -(k-1)\mu  &amp; \mu \\
\mu &amp; \mu &amp; \mu &amp; -(k-1)\mu \\
\end{pmatrix} \mbox{  ,}\]</span></p>
<p>Because this is a Jukes-Cantor-like model, state frequencies do not vary as a model parameter. These assumptions may seem unrealistic. However, all models are a compromise between reality and generalizability. Prior work has demonstrated that, in many conditions, the model does perform adequately. Because morphological characters do not carry common meaning across sites in a matrix in the way that nucleotide characters do, making assumptions that fit all characters is challenging. A visualization of this simple model can be seen in Fig. 1.</p>
<p>We will first perform a phylogenetic analysis using the Mk model. In further sections, we will explore how to relax key assumptions of the Mk model.</p>
</div>
<div id="tutorial-format" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorial-format" class="anchor"></a>Tutorial Format</h2>
<p>This tutorial follows a specific format for issuing instructions and information.</p>
<blockquote>
<p>The boxed instructions guide you to complete tasks that are not part of the RevBayes syntax, but rather direct you to create directories or files or similar.</p>
</blockquote>
<p>Information describing the commands and instructions will be written in paragraph-form before or after they are issued.</p>
<p>All command-line text, including all Rev syntax, are given in <code>monotype font</code>. Furthermore, blocks of Rev code that are needed to build the model, specify the analysis, or execute the run are given in separate shaded boxes. For example, we will instruct you to create a constant node called <code>example</code> that is equal to <code>1.0</code> using the <code>&lt;-</code> operator like this:</p>
<pre><code>example &lt;- 1.0</code></pre>
<p>If you copy and paste text from the browser, beware of introducing incorrect characters!</p>
</div>
<div id="data-and-files" class="section level2">
<h2 class="hasAnchor">
<a href="#data-and-files" class="anchor"></a>Data and Files</h2>
<blockquote>
<p>On your own computer, create a directory for the tutorial called <code>RB_DiscreteMorphology_Tutorial</code> (or whatever you want) and within this directory create a subdirectory called <strong>data</strong>. In this directory download the data files: <a href="vignettes/RB_Extended_morph/bears.nex"><code>bears.nex</code></a>.</p>
</blockquote>
</div>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting Started</h2>
<blockquote>
<p>Create a new subdirectory (in <strong>RB_Extended_morph_Tutorial</strong>) called <strong>scripts</strong>.</p>
</blockquote>
<p>When you execute RevBayes in this exercise, you will do so within the main directory you created (<strong>RB_Extended_morph_Tutorial</strong>), thus, if you are using a Unix-based operating system, we recommend that you add the RevBayes binary to your path. Alternatively make sure that you set the working directory to, for example, <strong>RB_Extended_morph_Tutorial</strong> if this is the directory you stored the scripts and data in.</p>
<p>In this section you will begin the file and write the Rev commands for loading in the taxon list and managing the data matrices. Then, starting in section Mk_model, you will move on to specifying each of the model components. Once the model specifications are complete, you will complete the script with the instructions given in section Complete_MCMC.</p>
</div>
<div id="load-data-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data-matrix" class="anchor"></a>Load Data Matrix</h2>
<p>RevBayes uses the function <code>readDiscreteCharacterData()</code> to load a data matrix to the workspace from a formatted file. This function can be used for both molecular sequences and discrete morphological characters. Import the morphological character matrix and assign it to the variable <code>morpho</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>morpho &lt;-<span class="st"> </span><span class="kw">readDiscreteCharacterData</span>(<span class="st">"data/bears.nex"</span>)</span></code></pre></div>
</div>
<div id="create-helper-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#create-helper-variables" class="anchor"></a>Create Helper Variables</h2>
<p>Before we begin writing the Rev scripts for each of the models, we need to instantiate a couple “helper variables” that will be used by downstream parts of our model specification.</p>
<p>Create a new constant node called <code>num_taxa</code> that is equal to the number of species in our analysis (18) and a constant node called <code>num_branches</code> representing the number of branches in the tree. We will also create a constant node of the taxon names. This list will be used to initialize the tree.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>taxa &lt;-<span class="st"> </span><span class="kw">morpho.names</span>()</span>
<span id="cb3-2"><a href="#cb3-2"></a>num_taxa &lt;-<span class="st"> </span><span class="kw">morpho.size</span>() </span>
<span id="cb3-3"><a href="#cb3-3"></a>num_branches &lt;-<span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>num_taxa <span class="op">-</span><span class="st"> </span><span class="dv">2</span></span></code></pre></div>
<p>Next, create two workspace variables called <code>moves</code> and <code>monitors</code>. These variables are vectors containing all of the MCMC moves and monitors, respectively.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>moves    =<span class="st"> </span><span class="kw">VectorMoves</span>()</span>
<span id="cb4-2"><a href="#cb4-2"></a>monitors =<span class="st"> </span><span class="kw">VectorMonitors</span>()</span></code></pre></div>
<p>One important distinction here is that <code>moves</code> and <code>monitors</code> are part of the RevBayes workspace and not the hierarchical model. Thus, we use the workspace assignment operator <code>=</code> instead of the constant node assignment <code>&lt;-</code>.</p>
</div>
<div id="the-mk-model-1" class="section level2">
<h2 class="hasAnchor">
<a href="#the-mk-model-1" class="anchor"></a>The Mk Model</h2>
<p>First, we will create a joint prior on the branch lengths and tree topology. In the first step we will specify a stochastic node for the mean of the</p>
<p>Some types of stochastic nodes can be updated by a number of alternative moves. Different moves may explore parameter space in different ways, and it is possible to use multiple different moves for a given parameter to improve mixing (the efficiency of the MCMC simulation). In the case of our unrooted tree topology, for example, we can use both a nearest-neighbor interchange move (<code>mvNNI</code>) and a subtree-prune and regrafting move (<code>mvSPR</code>). These moves do not have tuning parameters associated with them, thus you only need to pass in the <code>topology</code> node and proposal <code>weight</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>br_len_lambda <span class="op">~</span><span class="st"> </span><span class="kw">dnExp</span>(<span class="fl">0.2</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">moves.append</span>( <span class="kw">mvScale</span>(br_len_lambda, <span class="dt">weight=</span><span class="dv">2</span>) )</span>
<span id="cb5-3"><a href="#cb5-3"></a></span>
<span id="cb5-4"><a href="#cb5-4"></a>phylogeny <span class="op">~</span><span class="st"> </span><span class="kw">dnUniformTopologyBranchLength</span>(taxa, <span class="dt">branchLengthDistribution=</span><span class="kw">dnExp</span>(br_len_lambda))</span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="kw">moves.append</span>( <span class="kw">mvNNI</span>(phylogeny, <span class="dt">weight=</span>num_branches<span class="op">/</span><span class="fl">2.0</span>) )</span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="kw">moves.append</span>( <span class="kw">mvSPR</span>(phylogeny, <span class="dt">weight=</span>num_branches<span class="op">/</span><span class="fl">10.0</span>) )</span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="kw">moves.append</span>( <span class="kw">mvBranchLengthScale</span>(phylogeny, <span class="dt">weight=</span>num_branches) )</span>
<span id="cb5-8"><a href="#cb5-8"></a>    </span>
<span id="cb5-9"><a href="#cb5-9"></a>tree_length <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">phylogeny.treeLength</span>()</span></code></pre></div>
<p>Next, we will create a Q-matrix. Recall that the Mk model is simply a generalization of the JC model. Therefore, we will create a 2x2 Q-matrix using <code>fnJC</code>, which initializes Q-matrices with equal transition probabilities between all states.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>Q_morpho &lt;-<span class="st"> </span><span class="kw">fnJC</span>(<span class="dv">2</span>)</span></code></pre></div>
<p>Now that we have the basics of the model specified, we will add gamma-distributed rate variation and specify moves on the parameter of the gamma distribution, <code>alpha_morpho</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>alpha_morpho <span class="op">~</span><span class="st"> </span><span class="kw">dnUniform</span>( <span class="dv">0</span>, <span class="fl">1E8</span> )</span>
<span id="cb7-2"><a href="#cb7-2"></a>rates_morpho <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">fnDiscretizeGamma</span>( alpha_morpho, alpha_morpho, <span class="dv">4</span> )</span>
<span id="cb7-3"><a href="#cb7-3"></a></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co"># Moves on the parameters of the gamma distribution</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="kw">moves.append</span>( <span class="kw">mvScale</span>(alpha_morpho, <span class="dt">lambda=</span><span class="dv">1</span>, <span class="dt">weight=</span><span class="fl">2.0</span>)</span></code></pre></div>
<p>Lastly, we set up the CTMC. This should also be familiar from the {% page_ref ctmc %}. We see some familiar pieces: the tree, Q-matrix and site rates. We also have two new keywords: data type and coding. The data type argument specifies the type of data - in our case, “Standard”, the specification for morphology.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>phyMorpho <span class="op">~</span><span class="st"> </span><span class="kw">dnPhyloCTMC</span>(<span class="dt">tree=</span>phylogeny, <span class="dt">siteRates=</span>rates_morpho, <span class="dt">Q=</span>Q_morpho, <span class="dt">type=</span><span class="st">"Standard"</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">phyMorpho.clamp</span>(morpho)</span></code></pre></div>
<p>All of the components of the model are now specified.</p>
<p>Complete MCMC Analysis</p>
<p>Create Model Object</p>
<p>We can now create our workspace model variable with our fully specified model DAG. We will do this with the <code>model()</code> function and provide a single node from the graph (<code>phylogeny</code>).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>mymodel =<span class="st"> </span><span class="kw">model</span>(phylogeny)</span></code></pre></div>
<p>The object <code>mymodel</code> is a wrapper around the entire model graph and allows us to pass the model to various functions that are specific to our MCMC analysis.</p>
<p>Specify Monitors and Output Filenames</p>
<p>The next important step for our Rev-script is to specify the monitors and output file names. For this, we create a vector called <code>monitors</code> that will each sample and record or output our MCMC.</p>
<p>The first monitor we will create will monitor every named random variable in our model graph. This will include every stochastic and deterministic node using the <code>mnModel</code> monitor. The only parameter that is not included in the <code>mnModel</code> is the tree topology. Therefore, the parameters in the file written by this monitor are all numerical parameters. This file will be a tab-separated text file that can be opened by accessory programs for evaluating such parameters. We will also name the output file for this monitor and indicate that we wish to sample our MCMC every 10 cycles.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">monitors.append</span>( <span class="kw">mnModel</span>(<span class="dt">filename=</span><span class="st">"output/mk.log"</span>, <span class="dt">printgen=</span><span class="dv">10</span>) )</span></code></pre></div>
<p>The <code>mnFile</code> monitor writes any parameter we specify to file. Thus, if we only cared about the branch lengths and nothing else (this is not a typical or recommended attitude for an analysis this complex) we wouldn’t use the <code>mnModel</code> monitor above and just use the <code>mnFile</code> monitor to write a smaller and simpler output file. Since the tree topology is not included in the <code>mnModel</code> monitor (because it is not numerical), we will use <code>mnFile</code> to write the tree to file by specifying our <code>phylogeny</code> variable in the arguments.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">monitors.append</span>( <span class="kw">mnFile</span>(<span class="dt">filename=</span><span class="st">"output/mk.trees"</span>, <span class="dt">printgen=</span><span class="dv">10</span>, phylogeny) )</span></code></pre></div>
<p>The third monitor we will add to our analysis will print information to the screen.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="kw">monitors.append</span>( <span class="kw">mnScreen</span>(<span class="dt">printgen=</span><span class="dv">100</span>) )</span></code></pre></div>
</div>
<div id="set-up-the-mcmc" class="section level2">
<h2 class="hasAnchor">
<a href="#set-up-the-mcmc" class="anchor"></a>Set-Up the MCMC</h2>
<p>Once we have set up our model, moves, and monitors, we can now create the workspace variable that defines our MCMC run. We do this using the <code>mcmc()</code> function that takes the three main analysis components as arguments.</p>
<pre><code>mymcmc = mcmc(mymodel, monitors, moves, nruns=2, combine="mixed")</code></pre>
<p>The MCMC object that we named <code>mymcmc</code> has a member method called <code>.run()</code>. This will execute our analysis and we will set the chain length to <code>20000</code> cycles using the <code>generations</code> option.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">mymcmc.run</span>(<span class="dt">generations=</span><span class="dv">20000</span>, <span class="dt">tuningInterval=</span><span class="dv">200</span>)</span></code></pre></div>
<p>Once our Markov chain has terminated, we will want RevBayes to close. Tell the program to quit using the <code><a href="https://www.rdocumentation.org/packages/base/topics/quit">q()</a></code> function.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/quit">q</a></span>()</span></code></pre></div>
<blockquote>
<p>You made it! Save all of your files.</p>
</blockquote>
<p>##Execute the MCMC Analysis | subsec_run_MCMC %} With all the parameters specified and all analysis components in place, you are now ready to run your analysis. RevBayes cannot run inside the RStudio window. So we will need to copy our code out of RStudio. Luckily, the knitr package makes this very easy. Run the following in the console:</p>
<pre><code><a href="https://www.rdocumentation.org/packages/knitr/topics/knit">knitr::purl("vignettes/RB_Extended_morph/RB_MCMC_extendedMorph.Rmd")</a></code></pre>
<p>This will create a .R file containing our analysis script. Move it into the <code>scripts</code> folder and rename it to mk_gamma.Rev</p>
<blockquote>
<p>Begin by running the RevBayes executable. In Unix systems, type the following in your terminal (if the RevBayes binary is in your path): rb</p>
</blockquote>
<p>Provided that you started RevBayes from the correct directory, you can then use the <code><a href="https://www.rdocumentation.org/packages/base/topics/source">source()</a></code> function to feed RevBayes your Rev-script file (<code>mk_gamma.Rev</code>).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1"></a>    <span class="bu">source</span>(<span class="st">"scripts/mk_gamma.Rev"</span>)</span></code></pre></div>
<p>This will execute the analysis and you should see the following output (though not the exact same values):</p>
<pre><code>   Processing file "scripts/mcmc_mk.Rev"
   Successfully read one character matrix from file 'data/bears.nex'

   Running MCMC simulation
   This simulation runs 2 independent replicates.
   The simulator uses 5 different moves in a random move schedule with 58.4 moves per iteration

Iter        |      Posterior   |     Likelihood   |          Prior   |    elapsed   |        ETA   |
----------------------------------------------------------------------------------------------------
0           |       -680.054   |       -649.452   |       -30.6022   |   00:00:00   |   --:--:--   |
100         |       -419.885   |       -414.047   |       -5.83883   |   00:00:01   |   --:--:--   |
200         |       -427.028   |       -417.426   |       -9.60277   |   00:00:01   |   00:00:49   |
300         |       -421.585   |        -417.96   |        -3.6253   |   00:00:02   |   00:01:04   |
400         |       -431.561   |       -427.124   |       -4.43711   |   00:00:03   |   00:01:12   |
500         |       -423.507   |       -422.002   |       -1.50428   |   00:00:04   |   00:01:16   |
600         |       -418.061   |       -419.644   |        1.58298   |   00:00:05   |   00:01:18   |
700         |       -427.552   |       -423.884   |       -3.66793   |   00:00:05   |   00:01:06   |
800         |        -437.39   |       -424.302   |       -13.0876   |   00:00:06   |   00:01:09   |
900         |       -418.405   |       -413.872   |        -4.5323   |   00:00:07   |   00:01:10   |
1000        |       -425.641   |       -411.291   |       -14.3491   |   00:00:08   |   00:01:12   |

...</code></pre>
<p>When the analysis is complete, RevBayes will quit and you will have a new directory called <strong>output</strong> that will contain all of the files you specified with the monitors ({% ref subsubsec_Monitors %}).</p>
</div>
<div id="exercises" class="section level2">
<h2 class="hasAnchor">
<a href="#exercises" class="anchor"></a>Exercises</h2>
<ol style="list-style-type: decimal">
<li>Run the MCMC analysis in RevBayes.</li>
<li>Look at the resulting files <strong>mk_run_1.log</strong> and <strong>mk_run_2.log</strong> in Tracer and check for convergence.</li>
<li>Look at the majority rule consensus tree stored in <strong>mk.majrule.tre</strong> and the MAP tree stored in <strong>mk.map.tre</strong> in FigTree.</li>
</ol>
</div>
<div id="example-correcting-for-ascertainment-bias" class="section level2">
<h2 class="hasAnchor">
<a href="#example-correcting-for-ascertainment-bias" class="anchor"></a>Example: Correcting for Ascertainment Bias</h2>
<p>When Lewis first introduced the Mk model, he observed that branch lengths on the trees were greatly inflated. The reason for this is that when morphological characters are collected, characters that do not vary, or vary in a non-parsimony-informative way (such as autapomorphies) are excluded. Excluding these low-rate characters causes the overall amount of evolution to be over-estimated. This causes an inflation in the branch lengths {% cite Lewis2001 %}.</p>
<p>Therefore, when performing a morphological phylogenetic analysis, it is important to correct for this bias. There are numerous statistically valid ways to perform this correction {% cite allman08b %}. Original corrections simulated invariant and non-parsimony informative characters along the proposed tree. The likelihood of these characters would then be calculated and used to normalize the total likelihood value. RevBayes implements a dynamic programming approach that calculates the same likelihood, but does so faster.</p>
</div>
<div id="modifying-the-rev-script" class="section level2">
<h2 class="hasAnchor">
<a href="#modifying-the-rev-script" class="anchor"></a>Modifying the Rev-script</h2>
<blockquote>
<p>Create a copy of your previous Rev script, and call it <em>mcmc_Mkv.Rev</em>. You will need to modify the Rev. code provided in this section in this file.</p>
</blockquote>
<p>In RevBayes it is actually very simple to add a correction for ascertainment bias. You only need to set the option <code>coding="variable"</code> in the <code>dnPhyloCTMC</code>. Coding specifies what type of ascertainment bias is expected. We are using the <code>variable</code> correction, as we have no invariant character in our matrix. If we also lacked parsimony non-informative characters, we would use the coding <code>informative</code> (there are actually 4 parsimony non-informative characters in our matrix).</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>phyMorpho <span class="op">~</span><span class="st"> </span><span class="kw">dnPhyloCTMC</span>(<span class="dt">tree=</span>phylogeny, <span class="dt">siteRates=</span>rates_morpho, <span class="dt">Q=</span>Q_morpho, <span class="dt">type=</span><span class="st">"Standard"</span>, <span class="dt">coding=</span><span class="st">"variable"</span>)</span></code></pre></div>
<blockquote>
<p>Remember to change all filenames for the output, e.g., from <code>output/mk.log</code> to <code>output/mkv.log</code>.</p>
</blockquote>
<blockquote>
<p>That’s all you need to do! Now run this script in RevBayes.</p>
</blockquote>
</div>
</div>
<div id="example-relaxing-the-assumption-of-equal-transition-probabilities" class="section level1">
<h1 class="hasAnchor">
<a href="#example-relaxing-the-assumption-of-equal-transition-probabilities" class="anchor"></a>Example: Relaxing the Assumption of Equal Transition Probabilities</h1>
<p>The Mk model makes a number of assumptions, but one that may strike you as particularly unrealistic is the assumption that characters are equally likely to change from any one state to any other. That means that a trait is as likely to be gained as lost. While this may hold true for some traits, we expect that it may be untrue for many others.</p>
<p>RevBayes has functionality to allow us to relax this assumption. We do this by specifying a beta prior on state frequencies. Stationary frequencies impact how likely we are to see changes in a character. For example, it may be very likely, in a character, to change from 0 to 1. But if the frequency of 0 is very low, we will still seldom see this change.</p>
<p>We can exploit the relationship between state frequencies and observed changes to allow for variable Q-matrices across characters ({% ref morpho_graphical_model %}). To do this, we generate a beta distribution on state frequencies, and use the state frequencies from that distribution to generate a series of Q-matrices used to evaluate our data {% cite Pagel2004 %}.</p>
<p>This type of model is called a <strong>mixture model</strong>. There are assumed to be subdivisions in the data, which may require different parameters (in this case, state frequencies). These subdivisions are not defined <em>a priori</em>. This model has previously been shown to be effective for a range of empirical and simulated datasets {% cite Wright2016 %}.</p>
<p>{% figure morpho_graphical_model %} <img src="img/tikz/morpho_gm.png" width="400"> {% figcaption %} Graphical model demonstrating the discretized beta distribution for allowing variable state frequencies. {% endfigcaption %} {% endfigure %}</p>
<div id="modifying-the-rev-script-1" class="section level2">
<h2 class="hasAnchor">
<a href="#modifying-the-rev-script-1" class="anchor"></a>Modifying the Rev-script</h2>
<blockquote>
<p>Make a copy of the Rev script you made earlier. Call it <code>mcmc_mk_dicretized.Rev</code>. This new script will contain the new model parameters and models.</p>
</blockquote>
<p>We will use a discretized beta distribution to place a prior on the state frequencies. The beta distribution has two parameters, <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span>. These two parameters specify the shape of the distribution. State frequencies will be evaluated according to this distribution, in the same way that rate variation is evaluated according to the gamma distribution. The discretized distribution is split into multiple classes, each with it’s own set of frequencies for the 0 and 1 characters. The number of classes can vary; we have chosen 4 for tractability. Note that we need to make sure that this discretization results in a symmetric model, therefore we will use only one parameter for the beta distribution: <code>beta_scale</code> such that <span class="math inline">\(\alpha = \beta\)</span>.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>num_cats =<span class="st"> </span><span class="dv">4</span></span>
<span id="cb20-2"><a href="#cb20-2"></a>beta_scale <span class="op">~</span><span class="st"> </span><span class="kw">dnLognormal</span>( <span class="fl">0.0</span>, <span class="dt">sd=</span><span class="dv">2</span><span class="op">*</span><span class="fl">0.587405</span> )</span>
<span id="cb20-3"><a href="#cb20-3"></a><span class="kw">moves.append</span>( <span class="kw">mvScale</span>(beta_scale, <span class="dt">lambda=</span><span class="dv">1</span>, <span class="dt">weight=</span><span class="fl">5.0</span> ) )</span></code></pre></div>
<p>Above, we initialized the number of categories, the parameters of the beta distribution, and the moves on these parameters.</p>
<p>Next, we set the categories to each represent a quadrant of the beta distribution specified by <code>beta_scale</code>.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>cats <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">fnDiscretizeBeta</span>(beta_scale, beta_scale, num_cats)</span></code></pre></div>
<p>If you were to print the <code>cats</code> variable, you would see a list of state frequencies like so:</p>
<pre><code>[ 0.011, 0.236, 0.764, 0.989 ]</code></pre>
<p>Using these state frequencies, we will generate a new vector of Q-matrices. Because we are varying the state frequencies, we must use a Q-matrix generation function that allows for state frequencies to vary as a parameter. We will, therefore, use the <code>fnF81</code> function.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">cats.size</span>())</span>
<span id="cb23-2"><a href="#cb23-2"></a>{</span>
<span id="cb23-3"><a href="#cb23-3"></a>    Q[i] <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">fnF81</span>(<span class="kw">simplex</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(<span class="dv">1</span><span class="op">-</span>cats[i]), cats[i]))</span>
<span id="cb23-4"><a href="#cb23-4"></a>}</span></code></pre></div>
<p>Additionally, in RevBayes we need to specify the probabilities that a site evolves according to one of the Q-matrices. For this model the probabilities must be equal because we need to guarantee that the model is symmetric. Thus, we use a <code>simplex</code> function to create a vector that sums to 1.0.</p>
<pre><code>matrix_probs &lt;- simplex( rep(1,num_cats) )</code></pre>
<p>The only other specification that needs to change in the model specification is the CTMC:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>phyMorpho <span class="op">~</span><span class="st"> </span><span class="kw">dnPhyloCTMC</span>(<span class="dt">tree=</span>phylogeny, <span class="dt">siteRates=</span>rates_morpho, <span class="dt">Q=</span>Q, <span class="dt">type=</span><span class="st">"Standard"</span>, <span class="dt">coding=</span><span class="st">"variable"</span>, <span class="dt">siteMatrices=</span>matrix_probs)</span></code></pre></div>
<p>You will notice that we have added a command to tell the CTMC that we have multiple site matrices that will be applied to different characters in the matrix.</p>
<blockquote>
<p>Remember to change all filenames for the output, e.g., from <code>output/mkv.log</code> to <code>output/mkv_discretized.log</code>.</p>
</blockquote>
<blockquote>
<p>The MCMC chain set-up does not need to change. Run the new MCMC file, just as you ran the previous MCMC analyses. This estimation will take longer than the Mk model, due to increased model complexity.</p>
</blockquote>
</div>
</div>
<div id="evaluate-and-summarize-your-results" class="section level1">
<h1 class="hasAnchor">
<a href="#evaluate-and-summarize-your-results" class="anchor"></a>Evaluate and Summarize Your Results</h1>
<div id="evaluating-the-mcmc" class="section level3">
<h3 class="hasAnchor">
<a href="#evaluating-the-mcmc" class="anchor"></a>Evaluating the MCMC</h3>
<p>We will use <code>Tracer</code>to evaluate the MCMC samples from our three estimations. Load all three of the MCMC logs into the <code>Tracer</code> window. Highlight all three files in the upper left-hand viewer ({% ref add_files %}) by right- or command-clicking all three files.</p>
<p>Once all three trace logs are loaded and highlighted, first look at the estimated likelihoods. You will notice that the Mk model (<code>mk.log</code>), as originally proposed by Lewis is improved by adding the correction for ascertainment bias. The <code>mkv</code> and the <code>mkv_discretized</code> analyses both represent improvements, but are fairly close in likelihood score to each other. Likely, we would need to perform stepping stone model assessment to truly tell if the more complicated model is statistically justified. This analysis is too complicated and time-consuming for this tutorial period, but you will find instructions for performing the analysis in the Bayes Factor tutorial.</p>
<p>Next, click on the <code>Trace</code> tab. In the lower left hand corner, you will notice an option to color each trace by the file it came from. Choose this option (you may need to expand the window slightly to see it). Next to this option, you can also see an option to add a legend to your trace window. The results of this coloring can be seen in Tracer. You will see that the models are mixing quite well, but we may want to run the MCMC chains longer if we were to use these analyses in a paper.</p>
<p>We are interested in two aspects of the posterior distribution. First, the <code>mkv</code> and <code>mkv_discretized</code> analyses incorporated a correction for the biased sampling of variable characters, while the <code>mk</code> analysis does not. In this case, we expect the <code>tree_length</code> variable to be greater for the <code>mk</code> analysis, because in reality our data are enriched for variation (i.e. the model overestimates variation). Click on the “Marginal Density” tab and select the <code>tree_length</code> parameter. Tracer shows that <code>tree_length</code> is approximately 20% greater for the <code>mk</code>.</p>
<p>Second, we are interested in characterizing the degree of heterogeneity estimated by the beta discretized model. If the data were distributed by a single morphological rate matrix, then we would expect to see very little variation among the different <code>cats</code> value estimates, and a very large value for the <code>beta_scale</code> parameter of the discretized beta distribution. For example, if beta_scale = 1000, then that would cause all discretized beta categories to have values approaching 0.5, which approximates a symmetric Mk model. Examine the output for <code>beta_scale</code> in Tracer ({% ref tracer_alpha_beta %}).</p>
</div>
<div id="summarizing-trees" class="section level2">
<h2 class="hasAnchor">
<a href="#summarizing-trees" class="anchor"></a>Summarizing Trees</h2>
<p>The trees estimated in the above sections are summarized using a majority rule consensus tree (MRCT). Clades appearing in <span class="math inline">\(p&gt;0.5\)</span> of posterior samples are resolved in the MRCT, while poorly support clades with <span class="math inline">\(p \leq 0.5\)</span> are shown as unresolved polytomies. Poor phylogenetic resolution might be caused by having too few phylogenetically informative characters, or conflicting signals for certain species relationships. Let’s compare our topological estimates across models.</p>
<p>The MRCTs for the Mk model with and without the +v correction are very similar to that for the discretized-beta model ({% ref mk_discretized_majrule %}). Click on “Branch Labels” to show the branch lengths. Note that the branch lengths differ - the tree length is inflated without the +v correction, just as we saw when comparing the posterior tree length densities. In general, it is important to assess whether your results are sensitive to model assumptions, such as the degree of model complexity, and any mechanistic assumptions that motivate the model’s design. In this case, our tree estimate appears to be relatively robust to model complexity.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#graphical-models">Graphical Models</a></li>
      <li><a href="#overview-of-discrete-morphology-models">Overview of Discrete Morphology Models</a></li>
      <li><a href="#the-mk-model">The Mk Model</a></li>
      <li><a href="#tutorial-format">Tutorial Format</a></li>
      <li><a href="#data-and-files">Data and Files</a></li>
      <li><a href="#getting-started">Getting Started</a></li>
      <li><a href="#load-data-matrix">Load Data Matrix</a></li>
      <li><a href="#create-helper-variables">Create Helper Variables</a></li>
      <li><a href="#the-mk-model-1">The Mk Model</a></li>
      <li><a href="#set-up-the-mcmc">Set-Up the MCMC</a></li>
      <li><a href="#exercises">Exercises</a></li>
      <li><a href="#example-correcting-for-ascertainment-bias">Example: Correcting for Ascertainment Bias</a></li>
      <li><a href="#modifying-the-rev-script">Modifying the Rev-script</a></li>
      </ul>
</li>
      <li>
<a href="#example-relaxing-the-assumption-of-equal-transition-probabilities">Example: Relaxing the Assumption of Equal Transition Probabilities</a><ul class="nav nav-pills nav-stacked">
<li><a href="#modifying-the-rev-script-1">Modifying the Rev-script</a></li>
      </ul>
</li>
      <li>
<a href="#evaluate-and-summarize-your-results">Evaluate and Summarize Your Results</a><ul class="nav nav-pills nav-stacked">
<li><a href="#summarizing-trees">Summarizing Trees</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
