<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>treesifter advanced • apb2018</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="treesifter advanced">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">apb2018</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/RB_Discrete_Morphology/RB_Discrete_Morphology_Tutorial.html">RevBayes Discrete Morphology Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/RB_Total_Evidence/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/00-installs.html">Installs</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/01-treesiftr.html">treesifter</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
    <li>
      <a href="../../articles/R_exercises/03-FossilSim.html">FossilSim FBD Exercise</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>treesifter advanced</h1>
            
      
      
      <div class="hidden name"><code>02-treesiftr-advanced.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>treesiftr is an R package for visualizing the relationship between phylogenetic trees and phylogenetic data. Phylogenetic trees are crucial to the study of comparative biology, taxonomy, and evolution. However, understanding how to read a phylogenetic tree, and how a phylogenetic tree relates to underlying phylogenetic data, remain challenging.</p>
<p>In today’s lab exercise, we will learn about a phylogenetic matrix, and then use the data in the matrix to visualize a phylogenetic tree.</p>
<div id="the-fossil-bear-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#the-fossil-bear-matrix" class="anchor"></a>The fossil bear matrix</h2>
<p>The data matrix included with treesiftr is a matrix of binary (“0” and “1”) characters compiled to estimate a topology of living and extinct bear species <span class="citation">[@abella12]</span>. This matrix is fairly typical in size for a paleontological matrix, comprising 62 characters. It is, however, atypically complete, with only 18% missing data. In the following exercises, missing data will be represented by a thin black line. The “0” state will be represented in pale blue, and the “1” in brown.</p>
</div>
<div id="treesiftr" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr" class="anchor"></a>treesiftr</h2>
<p>treesiftr works by subsetting a phylogenetic matrix by user input. Then, a parsimony tree is constructed in Phangorn from the user-defined subset. The tree is scored under both parsimony and Lewis’ Mk model for discrete character data. The data and tree are then visualized using ggtree, based upon the ggplot2 package. This application makes use of Shiny to provide a graphical interface, but there is a second included tutorial for more experienced users of the R statistical language.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Currently, treesiftr can be installed via the devtools <code>install_github</code> function.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"wrightaprilm/treesiftr"</span>)</a></code></pre></div>
<pre><code>## Skipping install of 'treesiftr' from a github remote, the SHA1 (ec07434a) has not changed since last install.
##   Use `force = TRUE` to force installation</code></pre>
</div>
<div id="loading-required-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-required-packages" class="anchor"></a>Loading required packages</h2>
<p>treesiftr has a number of required packages. Install and load the below.</p>
</div>
<div id="subsetting-the-phylogenetic-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting-the-phylogenetic-matrix" class="anchor"></a>Subsetting the phylogenetic matrix</h2>
<p>The first step to making a treesiftr visualization is to select the subset of the phylogenetic matrix that we would like to visualize. This is performed via a function called <code>generate_sliding</code>. The below command will subset the</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">aln_path &lt;-<span class="st"> "data/bears_fasta.fa"</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">bears &lt;-<span class="st"> </span><span class="kw">read_alignment</span>(aln_path)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="st">"data/starting_tree.tre"</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">sample_df &lt;-<span class="st"> </span><span class="kw">generate_sliding</span>(bears, <span class="dt">start_char =</span> <span class="dv">1</span>, <span class="dt">stop_char =</span> <span class="dv">5</span>, <span class="dt">steps =</span> <span class="dv">1</span>)</a></code></pre></div>
<p>The result of this is a dataframe, shown below:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">sample_df</a></code></pre></div>
<pre><code>##   starting_val stop_val step_val
## 1            1        2        1
## 2            2        3        1
## 3            3        4        1
## 4            4        5        1
## 5            5        6        1</code></pre>
<p>This dataframe dispays the start character (the first character that will be visualized) and stop character (the final character that will be visualized).</p>
<p>We can then build trees from each subset:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw">generate_tree_vis</span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                         aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">output_vector</a></code></pre></div>
<p>Phangorn requires a starting tree to estimate a parsimony tree. We specify the tree we read in earlier for this purpose. <code>pscore</code> and <code>lscore</code> are optional arguments to view the parsimony and likelihood scores of the estimated trees. Remove these arguments before trying the exercises below.</p>
</div>
<div id="questions" class="section level2">
<h2 class="hasAnchor">
<a href="#questions" class="anchor"></a>Questions</h2>
<ol style="list-style-type: decimal">
<li>Visualize characters 1 and 2. What is the parsimony score for this character set. Add the parismony score to the visual like so:</li>
</ol>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw">generate_tree_vis</span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                         aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">                            <span class="dt">p_score =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">output_vector</a></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li><p>Visualize characters 2 and 3. What monophyletic group from tree 1 is no longer on this tree?</p></li>
<li><p>What is the parsimony score of the 31-34 character set? Click “Do you want to print the parsimony score?” in the infterface to check your answer.</p></li>
<li><p>Which character, 8, 9 or 10, represents a reveral?</p></li>
<li><p>What information would we need to decide if the “1” state possesed by <em>Zaragocyon_daamsi</em> is an autapomorphy?</p></li>
<li><p>Do all characters with the same parsimony score have the same likelihood score? You can add the likelihood score to the visualization like so:</p></li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw">generate_tree_vis</span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                         aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears, <span class="dt">pscore =</span> <span class="ot">TRUE</span>, <span class="dt">lscore =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">output_vector</a></code></pre></div>
<ol start="7" style="list-style-type: decimal">
<li><p>Compare characters 46-49 and 47-50. Why does set 47-50 have a better likelihood than 46-49?</p></li>
<li><p>What is the relationship between the likelihood score and increasing the number of characters visualized?</p></li>
<li><p>What is the minimum number adding a character can add to the parsimony score?</p></li>
<li><p>These trees are fully resolved. Based on your exploration of the data, does this degree of resolution make sense?</p></li>
</ol>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-fossil-bear-matrix">The fossil bear matrix</a></li>
      <li><a href="#treesiftr">treesiftr</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#loading-required-packages">Loading required packages</a></li>
      <li><a href="#subsetting-the-phylogenetic-matrix">Subsetting the phylogenetic matrix</a></li>
      <li><a href="#questions">Questions</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
