<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Dating a Phylogeny with the Fossilized Birth-Death • apb2019</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>



<meta property="og:title" content="Dating a Phylogeny with the Fossilized Birth-Death" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">apb2019</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="articles/RB_Discrete_Morphology/RB_MCMC_Discrete_Morph.html">Discrete morphology - Tree Inference</a>
    </li>
    <li>
      <a href="articles/RB_Total_Evidence/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="articles/R_exercises/00-installs.html">Installs</a>
    </li>
    <li>
      <a href="articles/R_exercises/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="articles/R_exercises/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
    <li>
      <a href="articles/R_exercises/03-FossilSim.html">FossilSim FBD Exercise</a>
    </li>
    <li>
      <a href="articles/lectures/TreeThinking.html">Tree Thinking</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Dating a Phylogeny with the Fossilized Birth-Death</h1>
    </div>


<div id="dating-a-phylogeny-with-the-fossilized-birth-death" class="section level1">

<p>author: April Wright date: 08.10.2018 autosize: true</p>
</div>
<div id="macroevolution-and-phylogeny" class="section level1">
<h1 class="hasAnchor">
<a href="#macroevolution-and-phylogeny" class="anchor"></a>Macroevolution and Phylogeny</h1>
<div id="phylogenetic-trees-are-often-used-to-study-patterns-of-diversification-trait-evolution-and-lineage-dispersal" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-trees-are-often-used-to-study-patterns-of-diversification-trait-evolution-and-lineage-dispersal" class="anchor"></a>Phylogenetic trees are often used to study patterns of diversification, trait evolution, and lineage dispersal</h2>
</div>
</div>
<div id="macroevolution-and-phylogeny-1" class="section level1">
<h1 class="hasAnchor">
<a href="#macroevolution-and-phylogeny-1" class="anchor"></a>Macroevolution and Phylogeny</h1>
<div id="phylogenetic-trees-are-often-used-to-study-patterns-of-diversification-trait-evolution-and-lineage-dispersal-1" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-trees-are-often-used-to-study-patterns-of-diversification-trait-evolution-and-lineage-dispersal-1" class="anchor"></a>Phylogenetic trees are often used to study patterns of diversification, trait evolution, and lineage dispersal</h2>
</div>
<div id="but-the-reverse-is-also-true" class="section level2">
<h2 class="hasAnchor">
<a href="#but-the-reverse-is-also-true" class="anchor"></a><em>But the reverse is also true</em>
</h2>
</div>
</div>
<div id="the-fossilized-birth-death-is-a-macroevolutionary-model" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-macroevolutionary-model" class="anchor"></a>The Fossilized Birth-Death is a Macroevolutionary Model</h1>
<div id="it-describes-the-distribution-of-lineages-through-time-stadler-2010" class="section level2">
<h2 class="hasAnchor">
<a href="#it-describes-the-distribution-of-lineages-through-time-stadler-2010" class="anchor"></a>It describes the distribution of lineages through time (Stadler 2010)</h2>
</div>
</div>
<div id="the-fossilized-birth-death-is-a-macroevolutionary-model-1" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-macroevolutionary-model-1" class="anchor"></a>The Fossilized Birth-Death is a Macroevolutionary Model</h1>
<ul>
<li>Several key parameters <img src="img/FBD_Simple.png">
</li>
</ul>
</div>
<div id="the-fossilized-birth-death-is-a-macroevolutionary-model-2" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-macroevolutionary-model-2" class="anchor"></a>The Fossilized Birth-Death is a Macroevolutionary Model</h1>
<ul>
<li>Several key parameters <img src="img/fbd_gm_lambda.png">
</li>
</ul>
</div>
<div id="the-fossilized-birth-death-is-a-macroevolutionary-model-3" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-macroevolutionary-model-3" class="anchor"></a>The Fossilized Birth-Death is a Macroevolutionary Model</h1>
<ul>
<li>Several key parameters <img src="img/fbd_gm_mu.png">
</li>
</ul>
</div>
<div id="the-fossilized-birth-death-is-a-macroevolutionary-model-4" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-macroevolutionary-model-4" class="anchor"></a>The Fossilized Birth-Death is a Macroevolutionary Model</h1>
<ul>
<li>Several key parameters <img src="img/fbd_gm_psi.png">
</li>
</ul>
</div>
<div id="the-fossilized-birth-death-is-a-macroevolutionary-model-5" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-macroevolutionary-model-5" class="anchor"></a>The Fossilized Birth-Death is a Macroevolutionary Model</h1>
<ul>
<li>Several key parameters <img src="img/fbd_gm_rho.png">
</li>
</ul>
</div>
<div id="the-fossilized-birth-death-is-a-phylogenetic-model" class="section level1">
<h1 class="hasAnchor">
<a href="#the-fossilized-birth-death-is-a-phylogenetic-model" class="anchor"></a>The Fossilized Birth-Death is a Phylogenetic Model</h1>
<ul>
<li>All of these parameters relate to the tree <img src="img/FBD_Simple.png">
</li>
</ul>
</div>
<div id="tree-models-hands-on-with-fossilsim" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-models-hands-on-with-fossilsim" class="anchor"></a>Tree Models: Hands-On with FossilSim</h1>
<p>vignettes/03-FossilSim</p>
<p>Look at section <code>Birth and Death</code> first.</p>
</div>
<div id="sampled-ancestors" class="section level1">
<h1 class="hasAnchor">
<a href="#sampled-ancestors" class="anchor"></a>Sampled Ancestors</h1>
<ul>
<li>Under the fossilized birth-death model, it is possible to sample lineages that also have descendants on the tree</li>
</ul>
<p><img src="img/foote.png"></p>
<p>When do we observe more sampled ancestors? Look at some of your tree plots to find out.</p>
<p>What do you think would happen to speciation rates if sampled ancestors are disallowed?</p>
</div>
<div id="tree-models-hands-on-with-fossilsim-1" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-models-hands-on-with-fossilsim-1" class="anchor"></a>Tree Models: Hands-On with FossilSim</h1>
<p>vignettes/03-FossilSim</p>
<p>Now look at the sampling section</p>
</div>
<div id="tree-models-hands-on-with-fossilsim-2" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-models-hands-on-with-fossilsim-2" class="anchor"></a>Tree Models: Hands-On with FossilSim</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(FossilSim)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ape)</span>
<span id="cb1-3"><a href="#cb1-3"></a>tree &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/FossilSim/topics/sim.fbd.taxa">sim.fbd.taxa</a></span>(<span class="dt">n=</span><span class="dv">10</span>, <span class="dt">numbsim=</span><span class="dv">1</span>, <span class="dt">lambda=</span><span class="dv">3</span>, <span class="dt">mu=</span><span class="dv">2</span>, <span class="dt">psi=</span><span class="dv">2</span>)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="DivTime-figure/unnamed-chunk-1-1.png" alt="plot of chunk unnamed-chunk-1"></p>
<p>This tree implies some values: that lineages have a time since origination, a time at extinction, and a time of sampling. How do we get them?</p>
</div>
<div id="yesterday-we-unveiled-a-conflation" class="section level1">
<h1 class="hasAnchor">
<a href="#yesterday-we-unveiled-a-conflation" class="anchor"></a>Yesterday, we unveiled a conflation</h1>
<p>Likelihood and Bayesian phylogenetic trees have a rate-based branch length</p>
</div>
<div id="distance--rate-x-time" class="section level1">
<h1 class="hasAnchor">
<a href="#distance--rate-x-time" class="anchor"></a>distance = rate x time</h1>
<p>What if we’re interested in time?</p>
<p><img src="img/stigallwright.png"></p>
<p>Stigall and Wright 2017</p>
</div>
<div id="why-are-we-interested-in-time" class="section level1">
<h1 class="hasAnchor">
<a href="#why-are-we-interested-in-time" class="anchor"></a>Why are we interested in time?</h1>
<ul>
<li>Time might be inherently interesting.</li>
</ul>
</div>
<div id="why-are-we-interested-in-time-1" class="section level1">
<h1 class="hasAnchor">
<a href="#why-are-we-interested-in-time-1" class="anchor"></a>Why are we interested in time?</h1>
<ul>
<li>Time might be inherently interesting</li>
<li>Time might tell us something interesting about natural history <img src="img/Crocs.png" alt="Alan Turner, 2016">
</li>
</ul>
<p>Turner, et al. 2016</p>
</div>
<div id="why-are-we-interested-in-time-2" class="section level1">
<h1 class="hasAnchor">
<a href="#why-are-we-interested-in-time-2" class="anchor"></a>Why are we interested in time?</h1>
<ul>
<li>Time might be inherently interesting</li>
<li>Time might tell us something interesting about natural history</li>
<li>Time might be required to test hypotheses</li>
</ul>
<p><img src="img/Alfaro.png" alt="Alfaro et al, 2018"> Alfaro et al. 2018</p>
</div>
<div id="information-in-a-phylogenetic-matrix" class="section level1">
<h1 class="hasAnchor">
<a href="#information-in-a-phylogenetic-matrix" class="anchor"></a>Information in a phylogenetic matrix</h1>
<ul>
<li>Does this tell you when (in absolute time) <em>Agriarctos spp</em> diverged from <em>Ailurarctos lufengensis</em>?</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(treesiftr)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(bears)</span>
<span id="cb2-3"><a href="#cb2-3"></a>bears[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</span></code></pre></div>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"

$Ballusia_elmensis
 [1] "?" "0" "?" "?" "?" "?" "0" "1" "?" "0" "0" "0" "0" "0" "0" "0" "0"
[18] "0" "?" "0" "1" "0" "0" "0" "0" "0" "?" "1" "0" "0" "0" "0" "1" "0"
[35] "0" "?" "?" "0" "?" "?" "?" "0" "0" "0" "0" "0" "1" "0" "1" "1" "1"
[52] "1" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?"</code></pre>
</div>
<div id="no-" class="section level1">
<h1 class="hasAnchor">
<a href="#no-" class="anchor"></a>No. </h1>
</div>
<div id="information-in-a-phylogenetic-matrix-1" class="section level1">
<h1 class="hasAnchor">
<a href="#information-in-a-phylogenetic-matrix-1" class="anchor"></a>Information in a phylogenetic matrix</h1>
<ul>
<li>Where do we get our information on absolute ages?</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(bears)</span>
<span id="cb4-2"><a href="#cb4-2"></a>bears[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</span></code></pre></div>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"

$Ballusia_elmensis
 [1] "?" "0" "?" "?" "?" "?" "0" "1" "?" "0" "0" "0" "0" "0" "0" "0" "0"
[18] "0" "?" "0" "1" "0" "0" "0" "0" "0" "?" "1" "0" "0" "0" "0" "1" "0"
[35] "0" "?" "?" "0" "?" "?" "?" "0" "0" "0" "0" "0" "1" "0" "1" "1" "1"
[52] "1" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?"</code></pre>
</div>
<div id="information-in-a-phylogenetic-matrix-2" class="section level1">
<h1 class="hasAnchor">
<a href="#information-in-a-phylogenetic-matrix-2" class="anchor"></a>Information in a phylogenetic matrix</h1>
<p>So, where do we get the information on absolute ages?</p>
</div>
<div id="information-in-a-phylogenetic-matrix-3" class="section level1">
<h1 class="hasAnchor">
<a href="#information-in-a-phylogenetic-matrix-3" class="anchor"></a>Information in a phylogenetic matrix</h1>
<p>So, where do we get the information on absolute ages?</p>
<div id="divergence-time-estimation-is-the-process-by-which-we-model-the-distribution-of-branch-rates-speciation--extinction-events-and-fossil-presence-to-obtain-node-ages-on-a-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#divergence-time-estimation-is-the-process-by-which-we-model-the-distribution-of-branch-rates-speciation--extinction-events-and-fossil-presence-to-obtain-node-ages-on-a-tree" class="anchor"></a>Divergence time estimation is the process by which we model the distribution of branch rates, speciation (&amp; extinction events) and fossil presence to obtain node ages on a tree</h2>
</div>
</div>
<div id="divergence-time-estimation" class="section level1">
<h1 class="hasAnchor">
<a href="#divergence-time-estimation" class="anchor"></a>Divergence time estimation</h1>
<p><img src="img/BayesianDivTime.png"></p>
<p>Image via Rachel Warnock</p>
</div>
<div id="divergence-time-estimation-1" class="section level1">
<h1 class="hasAnchor">
<a href="#divergence-time-estimation-1" class="anchor"></a>Divergence time estimation</h1>
<ul>
<li>Strict clock</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phangorn)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggtree)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</span>
<span id="cb6-4"><a href="#cb6-4"></a>aln_path &lt;-<span class="st"> 'data/bears_fasta.fa'</span></span>
<span id="cb6-5"><a href="#cb6-5"></a>sample_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/treesiftr/topics/generate_sliding">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">1</span>, <span class="dt">stop_char =</span> <span class="dv">2</span>, <span class="dt">steps =</span> <span class="dv">10</span>)</span>
<span id="cb6-6"><a href="#cb6-6"></a>output_vector &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/treesiftr/topics/generate_tree_vis">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                         aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears)</span>
<span id="cb6-7"><a href="#cb6-7"></a>output_vector[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="DivTime-figure/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4"></p>
<p>If we know the rate of sequence divergence is 1% per million years, we can date the tree by extrapolating from the amount of divergence.</p>
</div>
<div id="branch-rates" class="section level1">
<h1 class="hasAnchor">
<a href="#branch-rates" class="anchor"></a>Branch Rates</h1>
<ul>
<li>
<p>Strict clock</p>
<ul>
<li>Reality: the rate at which mutations arise varies based on generation time, lifestyle, DNA repair. Evolutionary forces cause differences in if a mutation is fixed, or removed from the population.</li>
</ul>
</li>
</ul>
</div>
<div id="branch-rates-1" class="section level1">
<h1 class="hasAnchor">
<a href="#branch-rates-1" class="anchor"></a>Branch Rates</h1>
<ul>
<li>Strict clock</li>
</ul>
<p><img src="img/morph_clock_gm-eps-converted-to.png"></p>
<p>Reality: the rate at which mutations arise varies based on generation time, lifestyle, DNA repair. Evolutionary forces cause differences in if a mutation is fixed, or removed from the population.</p>
</div>
<div id="branch-rates-2" class="section level1">
<h1 class="hasAnchor">
<a href="#branch-rates-2" class="anchor"></a>Branch Rates</h1>
<ul>
<li>Relaxed clock</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phytools)</span>
<span id="cb7-2"><a href="#cb7-2"></a>pb &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/phytools/topics/pbtree">pbtree</a></span>(<span class="dt">n =</span> <span class="dv">5</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a>branches &lt;-<span class="st"> </span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(pb<span class="op">$</span>tip.label)<span class="op">*</span><span class="dv">2</span>) <span class="op">-</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>rates &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Exponential">rexp</a></span>(<span class="dt">n =</span> branches, <span class="dt">rate =</span> <span class="dv">1</span>)</span>
<span id="cb7-5"><a href="#cb7-5"></a>pb<span class="op">$</span>edge.length &lt;-<span class="st"> </span>rates</span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/phytools/topics/plotBranchbyTrait">plotBranchbyTrait</a></span>(pb, rates, <span class="dt">mode =</span> <span class="st">"edges"</span>)</span></code></pre></div>
<p><img src="DivTime-figure/unnamed-chunk-5-1.png" alt="plot of chunk unnamed-chunk-5"></p>
</div>
<div id="branch-rates-3" class="section level1">
<h1 class="hasAnchor">
<a href="#branch-rates-3" class="anchor"></a>Branch Rates</h1>
<ul>
<li>Relaxed clock <img src="img/uexp_gm-eps-converted-to.png">
</li>
</ul>
</div>
<div id="branch-rates-4" class="section level1">
<h1 class="hasAnchor">
<a href="#branch-rates-4" class="anchor"></a>Branch Rates</h1>
<div id="many-ways-to-describe-how-branch-rates-are-distributed" class="section level2">
<h2 class="hasAnchor">
<a href="#many-ways-to-describe-how-branch-rates-are-distributed" class="anchor"></a>Many ways to describe how branch rates are distributed</h2>
<ul>
<li>Global clock (Zuckerkandl &amp; Pauling, 1962)</li>
<li>Uncorrelated/independent rates models (Drummond et al. 2006; Rannala &amp; Yang 2007; Lepage et al. 2007)</li>
<li>Log-normally distributed autocorrelated rates (Thorne, Kishino &amp; Painter 1998; Kishino, Thorne &amp; Bruno 2001; Thorne &amp; Kishino 2002)</li>
<li>Local clocks (Hasegawa, Kishino &amp; Yano 1989; Kishino &amp; Hasegawa 1990; Yoder &amp; Yang 2000; Yang &amp; Yoder 2003, Drummond and Suchard 2010)</li>
<li>Mixture models on branch rates (Heath, Holder, Huelsenbeck 2012)</li>
<li>Punctuated rate change model (Huelsenbeck, Larget and Swoford 2000)</li>
</ul>
<p>Slide from Rachel Warnock</p>
</div>
</div>
<div id="branch-rates-5" class="section level1">
<h1 class="hasAnchor">
<a href="#branch-rates-5" class="anchor"></a>Branch Rates</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>tree &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/FossilSim/topics/sim.fbd.taxa">sim.fbd.taxa</a></span>(<span class="dt">n=</span><span class="dv">10</span>, <span class="dt">numbsim=</span><span class="dv">1</span>, <span class="dt">lambda=</span><span class="dv">3</span>, <span class="dt">mu=</span><span class="dv">2</span>, <span class="dt">psi=</span><span class="dv">2</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tree[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="DivTime-figure/unnamed-chunk-6-1.png" alt="plot of chunk unnamed-chunk-6"></p>
<ul>
<li>Fundamental: In order to know something about the time since divergence, we need to know something about the rate at which differences accumulate</li>
</ul>
</div>
<div id="divergence-time-estimation-2" class="section level1">
<h1 class="hasAnchor">
<a href="#divergence-time-estimation-2" class="anchor"></a>Divergence Time Estimation</h1>
<p>So we’re done! We got it, right?</p>
</div>
<div id="divergence-time-estimation-3" class="section level1">
<h1 class="hasAnchor">
<a href="#divergence-time-estimation-3" class="anchor"></a>Divergence time estimation</h1>
<p><img src="img/BayesianDivTime.png"></p>
<p>Image via Rachel Warnock</p>
</div>
<div id="including-fossils" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils" class="anchor"></a>Including Fossils</h1>
<ul>
<li>Including fossils substantially improves estimations of divergence times
<ul>
<li>Fossils allow us to establish minimum ages for divergences</li>
</ul>
</li>
</ul>
<p><img src="img/tree_plot_with_fossils-eps-converted-to.png"></p>
</div>
<div id="including-fossils-calibrations" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-calibrations" class="anchor"></a>Including Fossils: Calibrations</h1>
<ul>
<li><p>Fossils have typically been incorporated into analyses as what are called node calibrations</p></li>
<li><p>Use distributions to describe the relationship between the fossil sampling event and the parent node (which generated the fossil)</p></li>
</ul>
</div>
<div id="including-fossils-calibrations-1" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-calibrations-1" class="anchor"></a>Including Fossils: Calibrations</h1>
<p><img src="img/sysbiosys032f01_ht.gif" alt="Heath Sys Bio 2012"></p>
<p>Heath 2012</p>
</div>
<div id="including-fossils-calibrations-2" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-calibrations-2" class="anchor"></a>Including Fossils: Calibrations</h1>
<p><img src="img/sysbiosys032f01_ht.gif" alt="Heath Sys Bio 2012"></p>
<ul>
<li>Phylogenetic position not estimated from data</li>
<li>We can only use the age of the oldest fossil</li>
</ul>
</div>
<div id="including-fossils-calibrations-3" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-calibrations-3" class="anchor"></a>Including Fossils: Calibrations</h1>
<p><img src="img/F1.large.jpg" alt="Warnock et al. 2012 Biol Letters"></p>
<p>Warnock et al. 2012</p>
</div>
<div id="including-fossils-calibrations-4" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-calibrations-4" class="anchor"></a>Including Fossils: Calibrations</h1>
<p><img src="img/F3.large.jpg" alt="Warnock et al. 2015 Biol Letters"></p>
<p>Warnock et al. 2015</p>
</div>
<div id="including-fossils-fossilized-birth-death" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-fossilized-birth-death" class="anchor"></a>Including Fossils: Fossilized Birth-Death</h1>
<p>Include fossils as tips in our analysis</p>
<p><img src="img/tree_plot_with_fossils-eps-converted-to.png"></p>
<p>Heath, 2014</p>
</div>
<div id="including-fossils-fossilized-birth-death-1" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-fossilized-birth-death-1" class="anchor"></a>Including Fossils: Fossilized Birth-Death</h1>
<p><img src="img/full_model_modular-eps-converted-to.png"></p>
</div>
<div id="including-fossils-fossilized-birth-death-2" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-fossilized-birth-death-2" class="anchor"></a>Including Fossils: Fossilized Birth-Death</h1>
<ul>
<li>Fossils can be placed simply as we did yesterday in the discrete morphology tutorial</li>
<li>Can also be placed via reversible jump MCMC to be sampled ancestors</li>
</ul>
<p><img src="img/tree_plot_with_fossils-eps-converted-to.png"></p>
</div>
<div id="including-fossils-fossilized-birth-death-3" class="section level1">
<h1 class="hasAnchor">
<a href="#including-fossils-fossilized-birth-death-3" class="anchor"></a>Including Fossils: Fossilized Birth-Death</h1>
<p>With a partner: describe this graphical model in a couple sentences. What does each piece do?</p>
<p><img src="img/GraphicalModel.png"></p>
<p>image from Rachel Warnock</p>
</div>


  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

