<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Research And Teaching With RevBayes • apb2019</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>



<meta property="og:title" content="Research And Teaching With RevBayes" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">apb2019</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="articles/RB_Discrete_Morphology/RB_MCMC_Discrete_Morph.html">Discrete morphology - Tree Inference</a>
    </li>
    <li>
      <a href="articles/RB_Total_Evidence/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="articles/R_exercises/00-installs.html">Installs</a>
    </li>
    <li>
      <a href="articles/R_exercises/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="articles/R_exercises/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
    <li>
      <a href="articles/R_exercises/03-FossilSim.html">FossilSim FBD Exercise</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Research And Teaching With RevBayes</h1>
    </div>


<div id="research-and-teaching-with-revbayes" class="section level1">

<p>author: date: autosize: true</p>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<ul>
<li>Introduction to graphical modeling</li>
<li>How graphical modeling makes phylogenetic estimation more flexible</li>
</ul>
</div>
<div id="the-graphical-modeling-framework" class="section level1">
<h1 class="hasAnchor">
<a href="#the-graphical-modeling-framework" class="anchor"></a>The Graphical Modeling Framework</h1>
<ul>
<li>What is a model?</li>
</ul>
</div>
<div id="the-graphical-modeling-framework-1" class="section level1">
<h1 class="hasAnchor">
<a href="#the-graphical-modeling-framework-1" class="anchor"></a>The Graphical Modeling Framework</h1>
<ul>
<li>What is a model?
<ul>
<li>A model uses mathematics to describe a process, or the behavior of a system</li>
<li>Important facets of the generating process of the data are written down as parameters</li>
</ul>
</li>
</ul>
</div>
<div id="the-graphical-modeling-framework-2" class="section level1">
<h1 class="hasAnchor">
<a href="#the-graphical-modeling-framework-2" class="anchor"></a>The Graphical Modeling Framework</h1>
<ul>
<li>What is a model?
<ul>
<li>A model uses mathematics to describe a process, or the behavior of a system</li>
<li>Important facets of the generating process of the data are written down as parameters</li>
</ul>
</li>
<li>A graphical model represents the relationships and dependencies between model paramters</li>
</ul>
</div>
<div id="a-simple-case" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-case" class="anchor"></a>A simple case</h1>
<ul>
<li>An archer: Let’s say the archer is a pretty good archer</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>good_shots &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Normal">rnorm</a></span>(<span class="dt">n=</span> <span class="dv">100</span>, <span class="dt">mean=</span> <span class="dv">0</span>, <span class="dt">sd =</span> <span class="dv">1</span>)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/hist">hist</a></span>(good_shots, <span class="dt">main=</span><span class="st">"Archer arrow shots"</span>,  <span class="dt">xlab =</span> <span class="st">"Inches from bullseye"</span>)</span></code></pre></div>
<p><img src="ResearchAndTeachingWithRevBayes-figure/unnamed-chunk-1-1.png" alt="plot of chunk unnamed-chunk-1"></p>
</div>
<div id="a-simple-case-1" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-case-1" class="anchor"></a>A simple case</h1>
<ul>
<li>An archer: Let’s say the archer is a pretty bad archer</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>bad_shots &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Normal">rnorm</a></span>(<span class="dt">n=</span> <span class="dv">100</span>, <span class="dt">mean=</span> <span class="dv">0</span>, <span class="dt">sd =</span> <span class="dv">3</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/hist">hist</a></span>(bad_shots, <span class="dt">main=</span><span class="st">"Archer arrow shots"</span>,  <span class="dt">xlab =</span> <span class="st">"Inches from bullseye"</span>)</span></code></pre></div>
<p><img src="ResearchAndTeachingWithRevBayes-figure/unnamed-chunk-2-1.png" alt="plot of chunk unnamed-chunk-2"></p>
</div>
<div id="a-simple-case-2" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-case-2" class="anchor"></a>A simple case</h1>
<ul>
<li>It’s easy to describe a model where we are manipulating one parameter</li>
<li>But we rarely have such a simple case when we’re analysing biological data</li>
</ul>
</div>
<div id="a-not-so-simple-case" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>It’s easy to describe a model where we are manipulating one parameter</li>
<li>But we rarely have such a simple case when we’re analysing biological data</li>
<li>We’re describing the mechanisms that generated our observed data
<ul>
<li>In the case of a tree, this is the process of molecular or morphological evolution</li>
</ul>
</li>
</ul>
</div>
<div id="a-not-so-simple-case-1" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-1" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>This is where graphical models come in</li>
</ul>
</div>
<div id="a-not-so-simple-case-2" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-2" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>This is where graphical models come in</li>
<li>A <strong>graphical model</strong> represents the joint probability distribution as a graph</li>
</ul>
</div>
<div id="a-not-so-simple-case-3" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-3" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li>
<strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes.</li>
</ul>
</div>
<div id="a-not-so-simple-case-4" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-4" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li>
<strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes.</li>
</ul>
</div>
<div id="a-not-so-simple-case-5" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-5" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li>
<strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes. <img src="img/tikz/Normal.tex.png" alt="Image Title">
</li>
</ul>
</div>
<div id="a-not-so-simple-case-6" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-6" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li>
<strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes. <img src="img/tikz/Normal.tex.png" alt="Image Title">
</li>
</ul>
<p>Pr( data | model )</p>
</div>
<div id="a-not-so-simple-case-7" class="section level1">
<h1 class="hasAnchor">
<a href="#a-not-so-simple-case-7" class="anchor"></a>A <strong>not so</strong> simple case</h1>
<p><img src="img/tikz/graphical_model_legend.png" alt="Image Title"></p>
</div>
<div id="phylogenetical-complexity" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetical-complexity" class="anchor"></a>Phylogenetical Complexity</h1>
<ul>
<li>This is where graphical models come in</li>
<li>A graphical model represents the joint probability distribution as a graph</li>
<li>
<strong>Nodes</strong> are variables. <strong>Edges</strong> represent dependencies between nodes. <img src="img/tikz/Mk_model.png" alt="Image Title">
</li>
</ul>
</div>
<div id="phylogenetical-complexity-1" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetical-complexity-1" class="anchor"></a>Phylogenetical complexity</h1>
<p><img src="img/tikz/constant_nodes.png" alt="Image Title"></p>
</div>
<div id="phylogenetical-complexity-2" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetical-complexity-2" class="anchor"></a>Phylogenetical complexity</h1>
<p><img src="img/tikz/stoch_nodes.png" alt="Image Title"></p>
</div>
<div id="phylogenetical-complexity-3" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetical-complexity-3" class="anchor"></a>Phylogenetical complexity</h1>
<p><img src="img/tikz/deterministic_node.tex.png" alt="Image Title"></p>
</div>
<div id="phylogenetical-complexity-4" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetical-complexity-4" class="anchor"></a>Phylogenetical complexity</h1>
<p><img src="img/tikz/observed_node.tex.png" alt="Image Title"></p>
</div>
<div id="phylogenetical-complexity-5" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetical-complexity-5" class="anchor"></a>Phylogenetical complexity</h1>
<p><img src="img/tikz/gtrg_graphical_model.png" alt="Image Title"></p>
</div>
<div id="who-cares-about-graphical-modeling" class="section level1">
<h1 class="hasAnchor">
<a href="#who-cares-about-graphical-modeling" class="anchor"></a>Who cares about graphical modeling?</h1>
<p>I’m going to make the case that this framework enables radical transparency in both research and pedagogy.</p>
</div>
<div id="engaging-with-underlying-assumptions" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions" class="anchor"></a>Engaging with underlying assumptions</h1>
<ul>
<li>Most phylogenetic estimations treat among-site rate variation as Gamma-distributed</li>
<li>Not every site in a nucleotide alignment evolves at the same rate</li>
<li>Not every character in a morphological dataset evolves at the same rate</li>
</ul>
</div>
<div id="engaging-with-underlying-assumptions-1" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-1" class="anchor"></a>Engaging with underlying assumptions</h1>
<p><img src="img/journal.pone.0109210.g003.png"></p>
</div>
<div id="engaging-with-underlying-assumptions-2" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-2" class="anchor"></a>Engaging with underlying assumptions</h1>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</span>
<span id="cb3-3"><a href="#cb3-3"></a>dat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">draws =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/GammaDist">rgamma</a></span>(<span class="dt">n =</span> <span class="dv">500</span>, <span class="dt">shape =</span> <span class="fl">.1</span>, <span class="dt">scale =</span> <span class="dv">10</span>),<span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/GammaDist">rgamma</a></span>(<span class="dt">n =</span> <span class="dv">500</span>, <span class="dt">shape =</span> <span class="fl">.5</span>, <span class="dt">scale =</span> <span class="dv">10</span>)))</span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a>d2 &lt;-<span class="st"> </span>dat <span class="op">%&gt;%</span><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span>(<span class="dt">lower =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.25</span>), <span class="dt">middle =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.5</span>), <span class="dt">seventyfive =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.75</span>), <span class="dt">upper =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.99</span>))</span>
<span id="cb3-6"><a href="#cb3-6"></a></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(dat, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x=</span>draws)) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> lower)) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> middle)) <span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> seventyfive)) <span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> upper)) </span></code></pre></div>
<p><img src="ResearchAndTeachingWithRevBayes-figure/unnamed-chunk-3-1.png" alt="plot of chunk unnamed-chunk-3"></p>
</div>
<div id="engaging-with-underlying-assumptions-3" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-3" class="anchor"></a>Engaging with underlying assumptions</h1>
<p><img src="img/lnl.png" alt="Yang1994"></p>
<p>Yang (1994) demonstrated that modeling ASRV as Gamma distributed with 4 Gamma categories is sufficient</p>
</div>
<div id="engaging-with-underlying-assumptions-4" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-4" class="anchor"></a>Engaging with underlying assumptions</h1>
<p><img src="img/Gamma.png" alt="Yang1994"></p>
<p>Implies many low- or zero-rate characters</p>
</div>
<div id="engaging-with-underlying-assumptions-5" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-5" class="anchor"></a>Engaging with underlying assumptions</h1>
<p><img src="img/Fig5.png" alt="Ants5"></p>
<ul>
<li>We typically don’t collect invarient characters!<br>
</li>
<li>Wagner (2009) and Harisson and Larson (2016) have argued that the Gamma is not appropriate for ASRV in morphological data, and we should use a lognormal</li>
</ul>
</div>
<div id="engaging-with-underlying-assumptions-6" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-6" class="anchor"></a>Engaging with underlying assumptions</h1>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</span>
<span id="cb4-3"><a href="#cb4-3"></a>dat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">draws =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Lognormal">rlnorm</a></span>(<span class="dt">n =</span> <span class="dv">100</span>, <span class="dt">mean=</span><span class="dv">1</span>)))</span>
<span id="cb4-4"><a href="#cb4-4"></a></span>
<span id="cb4-5"><a href="#cb4-5"></a>d2 &lt;-<span class="st"> </span>dat <span class="op">%&gt;%</span><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span>(<span class="dt">lower =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.25</span>), <span class="dt">middle =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.5</span>), <span class="dt">seventyfive =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.75</span>), <span class="dt">upper =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/quantile">quantile</a></span>(dat<span class="op">$</span>draws, <span class="dt">probs =</span> <span class="fl">.99</span>))</span>
<span id="cb4-6"><a href="#cb4-6"></a></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(dat, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x=</span>draws)) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> lower)) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> middle)) <span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> seventyfive)) <span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">data =</span> d2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xintercept =</span> upper)) </span></code></pre></div>
<p><img src="ResearchAndTeachingWithRevBayes-figure/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4"></p>
</div>
<div id="engaging-with-underlying-assumptions-7" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-7" class="anchor"></a>Engaging with underlying assumptions</h1>
<ul>
<li>In previous iterations of software, you might have to email a developer and ask them to implement something</li>
<li>RevBayes’ graphical model framework emphasizes flexibility. We can simply plug in another ASRV distribution.</li>
</ul>
</div>
<div id="engaging-with-underlying-assumptions-8" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-8" class="anchor"></a>Engaging with underlying assumptions</h1>
<p>Open RB_Discrete_Morphology_Model_test.md</p>
</div>
<div id="engaging-with-underlying-assumptions-9" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-9" class="anchor"></a>Engaging with underlying assumptions</h1>
<p>RevBayes can’t be run from within RStudio. In our console, type:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(knitr)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/knit">purl</a></span>(<span class="st">"RB_MCMC_Discrete_Morph.Rmd"</span>)</span></code></pre></div>
<p>This will generate a RB_MCMC_Discrete_Morph.R file with our Rev code.</p>
</div>
<div id="engaging-with-underlying-assumptions-10" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-10" class="anchor"></a>Engaging with underlying assumptions</h1>
<p>Move the RB_MCMC_Discrete_Morph.R file to scripts, and execute like so:</p>
<pre><code>rb scripts/RB_MCMC_Discrete_Morph.R</code></pre>
<p>This will take about 7 minutes.</p>
</div>
<div id="engaging-with-underlying-assumptions-11" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-11" class="anchor"></a>Engaging with underlying assumptions</h1>
<p>Return to RB_MCMC_Discrete_Morph.md and scroll to Lognormally-distributed among-character rate variation.</p>
<p>Follow the instructions in the file.</p>
</div>
<div id="engaging-with-underlying-assumptions-12" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-12" class="anchor"></a>Engaging with underlying assumptions</h1>
</div>
<div id="engaging-with-underlying-assumptions-13" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-13" class="anchor"></a>Engaging with underlying assumptions</h1>
</div>
<div id="engaging-with-underlying-assumptions-14" class="section level1">
<h1 class="hasAnchor">
<a href="#engaging-with-underlying-assumptions-14" class="anchor"></a>Engaging with underlying assumptions</h1>
<ul>
<li>Maybe there is an effect here</li>
<li>Dan will show us some better ways of looking at posterior samples</li>
<li>But, the larger point is that <em>flexibility enabled by modularity means that we were able to immediately implement a model we were interested in</em>
</li>
</ul>
</div>
<div id="transparency-and-documentation" class="section level1">
<h1 class="hasAnchor">
<a href="#transparency-and-documentation" class="anchor"></a>Transparency and documentation</h1>
<p>It wouldn’t be unfair to say there are some things about RevBayes that are more complex than other phylogenetic softwares</p>
</div>
<div id="transparency-and-documentation-1" class="section level1">
<h1 class="hasAnchor">
<a href="#transparency-and-documentation-1" class="anchor"></a>Transparency and documentation</h1>
<p><img src="img/RB_tutorial.png" alt="tutorials"></p>
</div>
<div id="transparency-and-documentation-2" class="section level1">
<h1 class="hasAnchor">
<a href="#transparency-and-documentation-2" class="anchor"></a>Transparency and documentation</h1>
<p><img src="img/RB_tutorial.png" alt="tutorials"></p>
<p>Each one is available as a markdown file on the <a href="https://github.com/revbayes/revbayes.github.io">website repository</a></p>
</div>
<div id="transparency-and-documentation-3" class="section level1">
<h1 class="hasAnchor">
<a href="#transparency-and-documentation-3" class="anchor"></a>Transparency and documentation</h1>
<p>You see an analysis you like? Clone it!</p>
</div>
<div id="transparency-and-documentation-4" class="section level1">
<h1 class="hasAnchor">
<a href="#transparency-and-documentation-4" class="anchor"></a>Transparency and documentation</h1>
<p>Integrate with various other R packages, like Bookdown, to generate a flexible living textbook for your systematics class!</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>bookdown<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/bookdown/topics/render_book">render_book</a></span>(<span class="st">"RB_MCMC_Discrete_Morph.Rmd"</span>, <span class="st">"bookdown::gitbook"</span>)</span></code></pre></div>
</div>
<div id="transparency-and-documentation-5" class="section level1">
<h1 class="hasAnchor">
<a href="#transparency-and-documentation-5" class="anchor"></a>Transparency and documentation</h1>
<p>We also have the RevKernel, and a set of RevNotebooks ready to go for the Pythonistas.</p>
</div>


  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

