<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Tree Thinking • apb2018</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>



<meta property="og:title" content="Tree Thinking" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">apb2018</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="articles/RB_Discrete_Morphology/RB_Discrete_Morphology_Tutorial.html">RevBayes Discrete Morphology Tree Estimation</a>
    </li>
    <li>
      <a href="articles/RB_Total_Evidence/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="articles/R_exercises/00-installs.html">Installs</a>
    </li>
    <li>
      <a href="articles/R_exercises/01-treesiftr.html">treesifter</a>
    </li>
    <li>
      <a href="articles/R_exercises/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
    <li>
      <a href="articles/R_exercises/03-FossilSim.html">FossilSim FBD Exercise</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Tree Thinking</h1>
    </div>


<div id="tree-thinking" class="section level1">

<p>author: April Wright date: 08.09.2018 autosize: true</p>
</div>
<div id="good-morning" class="section level1">
<h1 class="hasAnchor">
<a href="#good-morning" class="anchor"></a>Good Morning!</h1>
<ul>
<li>What is a tree?</li>
<li>How is a tree built?</li>
<li>What are phylogenetic data?</li>
</ul>
</div>
<div id="what-do-we-do-with-a-phylogeny" class="section level1">
<h1 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny" class="anchor"></a>What do we do with a phylogeny?</h1>
<ul>
<li>Determine the timing of trait evolution</li>
</ul>
<p><img src="img/skinks.png" alt="Skink tree from Wright et al. 2015"></p>
</div>
<div id="what-do-we-do-with-a-phylogeny-1" class="section level1">
<h1 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-1" class="anchor"></a>What do we do with a phylogeny?</h1>
<p>-Tell homology from convergence</p>
<p><img src="img/A_spinner_dolphin_in_the_Red_Sea.jpg" alt="Dolphin, Alex Vasenin via WikiMedia"><img src="img/skeleton.jpg" alt="Dolphin"></p>
</div>
<div id="what-do-we-do-with-a-phylogeny-2" class="section level1">
<h1 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-2" class="anchor"></a>What do we do with a phylogeny?</h1>
<p>-Trace the origins of structures</p>
<p><img src="img/AskABiologist.png" alt="Ask a Biologist"></p>
</div>
<div id="what-do-we-do-with-a-phylogeny-3" class="section level1">
<h1 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-3" class="anchor"></a>What do we do with a phylogeny?</h1>
<p>-Taxonomy</p>
<ul>
<li>Hennig, 1950 Grundzüge einer Theorie der Phylogenetischen Systematik
<ul>
<li>Taxonomy should be logically consistent with the tree for the group</li>
</ul>
</li>
</ul>
</div>
<div id="what-do-we-do-with-a-phylogeny-4" class="section level1">
<h1 class="hasAnchor">
<a href="#what-do-we-do-with-a-phylogeny-4" class="anchor"></a>What do we do with a phylogeny?</h1>
<p>-Taxonomy</p>
<ul>
<li>Hennig, 1950 Grundzüge einer Theorie der Phylogenetischen Systematik
<ul>
<li>Taxonomy should be logically consistent with the tree for the group</li>
</ul>
</li>
<li>Sneath &amp; Sokal, 1963, 1973
<ul>
<li>Using distance matrices to cluster based on phenetic similarity</li>
</ul>
</li>
</ul>
</div>
<div id="tree-terms-tip" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-tip" class="anchor"></a>Tree Terms: Tip</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(phytools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">tree &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/phytools/topics/pbtree">pbtree</a></span>(<span class="dt">n =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-1-1.png" alt="plot of chunk unnamed-chunk-1"> Tip: What we are putting on the tree. May be species, individuals, or higher-order taxa. May be called terminal node, leaf, one degree node.</p>
</div>
<div id="tree-terms-tip-1" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-tip-1" class="anchor"></a>Tree Terms: Tip</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(phytools)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">tree &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/phytools/topics/pbtree">pbtree</a></span>(<span class="dt">n =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-2-1.png" alt="plot of chunk unnamed-chunk-2"> Branch: What connects the tip to the tree. Can have a variety of units, which we will discuss over the next few days. May be called edge.</p>
</div>
<div id="tree-terms-node" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-node" class="anchor"></a>Tree Terms: Node</h1>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(phytools)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tree &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/phytools/topics/pbtree">pbtree</a></span>(<span class="dt">n =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">nodelabels</span>()</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-3-1.png" alt="plot of chunk unnamed-chunk-3"> Node: Where nodes meet, implying a most recent common ancestor. May be called vertex, or three-degree node.</p>
</div>
<div id="tree-terms" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms" class="anchor"></a>Tree Terms</h1>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>, <span class="dt">direction =</span> <span class="st">"downwards"</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4"></p>
</div>
<div id="tree-terms-1" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-1" class="anchor"></a>Tree Terms</h1>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>, <span class="dt">type=</span><span class="st">"fan"</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-5-1.png" alt="plot of chunk unnamed-chunk-5"></p>
</div>
<div id="tree-terms-rotation---reflecting-taxa-at-a-node" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-rotation---reflecting-taxa-at-a-node" class="anchor"></a>Tree Terms: Rotation - reflecting taxa at a node</h1>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">nodelabels</span>(<span class="dt">cex =</span> <span class="fl">3.5</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-6-1.png" alt="plot of chunk unnamed-chunk-6"></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co">#rotateNodes(tree, c(7, 8))</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#plot(tree, cex = 3.5, no.margin = TRUE, edge.width = 1.5)</span></a></code></pre></div>
</div>
<div id="tree-terms-monophyletic---an-ancestor-and-all-its-descendents" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-monophyletic---an-ancestor-and-all-its-descendents" class="anchor"></a>Tree Terms: Monophyletic - an ancestor and all its descendents</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">is.monophyletic</span>(tree, <span class="kw">c</span>(<span class="st">"t1"</span>, <span class="st">"t2"</span>), <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-7-1.png" alt="plot of chunk unnamed-chunk-7"></p>
<pre><code>[1] FALSE</code></pre>
</div>
<div id="tree-terms-rooting" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-rooting" class="anchor"></a>Tree Terms: Rooting</h1>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co"># reroot(tree, node.number)</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw">plot</span>(tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-8-1.png" alt="plot of chunk unnamed-chunk-8"></p>
<p>Ingroup: Taxa of interest</p>
<p>Outgroup: Taxon closely related used to root the tree</p>
</div>
<div id="tree-terms-rooting-1" class="section level1">
<h1 class="hasAnchor">
<a href="#tree-terms-rooting-1" class="anchor"></a>Tree Terms: Rooting</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">unroot_tree &lt;-<span class="st"> </span><span class="kw">unroot</span>(tree)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">plot</span>(unroot_tree, <span class="dt">cex =</span> <span class="fl">3.5</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">edge.width =</span> <span class="fl">1.5</span>)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-9-1.png" alt="plot of chunk unnamed-chunk-9"></p>
</div>
<div id="how-is-a-tree-built" class="section level1">
<h1 class="hasAnchor">
<a href="#how-is-a-tree-built" class="anchor"></a>How is a tree built?</h1>
<ul>
<li>Many ways. We will focus on three:
<ul>
<li>Maximum parsimony</li>
<li>Maximum likelihood</li>
<li>Bayesian inference</li>
</ul>
</li>
</ul>
</div>
<div id="phylogenetic-data" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data" class="anchor"></a>Phylogenetic Data</h1>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">library</span>(alignfigR)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">char_data &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(<span class="st">"data/bears_fasta.fa"</span>)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">char_data[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
</div>
<div id="phylogenetic-data-1" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-1" class="anchor"></a>Phylogenetic Data</h1>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">library</span>(alignfigR)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">char_data &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(<span class="st">"data/bears_fasta.fa"</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">char_data[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<p>These data are binary</p>
</div>
<div id="phylogenetic-data-2" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-2" class="anchor"></a>Phylogenetic Data</h1>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">library</span>(alignfigR)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">char_data &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(<span class="st">"data/bears_fasta.fa"</span>)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">char_data[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>$Agriarctos_spp
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "0" "1" "?" "1" "1" "?" "0" "1" "1" "1"
[35] "1" "0" "1" "1" "0" "?" "?" "0" "1" "1" "1" "0" "?" "?" "?" "?" "?"
[52] "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailurarctos_lufengensis
 [1] "?" "0" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?" "?"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "1" "?" "1" "1" "?" "0" "?" "?" "?"
[35] "?" "0" "1" "1" "1" "?" "0" "0" "1" "1" "1" "0" "1" "0" "1" "1" "0"
[52] "1" "1" "?" "?" "?" "?" "?" "?" "?" "?" "?"

$Ailuropoda_melanoleuca
 [1] "1" "0" "1" "1" "1" "1" "0" "1" "1" "0" "1" "0" "0" "1" "0" "0" "0"
[18] "0" "0" "1" "1" "1" "1" "0" "1" "0" "1" "1" "1" "0" "0" "1" "0" "1"
[35] "0" "0" "1" "1" "0" "0" "0" "0" "1" "1" "1" "0" "1" "0" "0" "1" "0"
[52] "1" "1" "0" "0" "0" "1" "0" "0" "0" "1" "0"</code></pre>
<p>Always arranged with rows being taxa and columns corresponding to a character - “matrix” structure</p>
</div>
<div id="phylogenetic-data-3" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-3" class="anchor"></a>Phylogenetic Data</h1>
<p>Example character from Brady:</p>
<ol>
<li>Worker, queen, and male. Specialized, stout setae on anterior margin of clypeus: (0) absent; (1) present. The presence of these specialized setae is a putative synapomorphy of Amblyoponinae (Ward, 1994), including Amblyopone and Onychomyrmex. <img src="img/2000px-Front_of_insect_head_diagram.svg.png">
</li>
</ol>
</div>
<div id="phylogenetic-data-4" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-4" class="anchor"></a>Phylogenetic Data</h1>
<ul>
<li>How do we know we have a truly discrete state? <img src="img/2000px-Front_of_insect_head_diagram.svg.png">
</li>
</ul>
</div>
<div id="phylogenetic-data-5" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-5" class="anchor"></a>Phylogenetic Data</h1>
<p>How do we know we’ve captured the relevant character axes?</p>
<p><img src="img/butterfly_human_vision.jpg" alt="Ask a biologist, Mike Hagelberg"></p>
</div>
<div id="phylogenetic-data-6" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-6" class="anchor"></a>Phylogenetic Data</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">colors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"blue"</span>, <span class="st">"purple"</span>,<span class="st">"white"</span>)</a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/plot_alignment">plot_alignment</a></span>(char_data, colors, <span class="dt">taxon_labels =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme">theme</a></span>(<span class="dt">text =</span> <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/element">element_text</a></span>(<span class="dt">size=</span><span class="dv">40</span>))</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-13-1.png" alt="plot of chunk unnamed-chunk-13"></p>
</div>
<div id="phylogenetic-data-7" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-data-7" class="anchor"></a>Phylogenetic Data</h1>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">colors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"blue"</span>, <span class="st">"purple"</span>,<span class="st">"white"</span>)</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/plot_alignment">plot_alignment</a></span>(char_data, colors, <span class="dt">taxon_labels =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme">theme</a></span>(<span class="dt">text =</span> <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/element">element_text</a></span>(<span class="dt">size=</span><span class="dv">40</span>))</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-14-1.png" alt="plot of chunk unnamed-chunk-14"> How do we go from this to a tree?</p>
</div>
<div id="parsimony" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony" class="anchor"></a>Parsimony</h1>
<ul>
<li>Not only applied in phylogenetics</li>
<li>The simplest explanation for the observed data is the best</li>
</ul>
</div>
<div id="parsimony-1" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony-1" class="anchor"></a>Parsimony</h1>
<ul>
<li>Maximum parsimony: the tree that minimizes the number of “steps”, or changes, on a tree is to be preferred</li>
<li>Let’s turn to the board for a minute: Parsimony informative, invariant, and parsimony non-informative variation</li>
</ul>
<p>??? Have them start installs on the next page while we do this.</p>
</div>
<div id="treesiftr" class="section level1">
<h1 class="hasAnchor">
<a href="#treesiftr" class="anchor"></a>treesiftr</h1>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">library</span>(treesiftr)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">aln_path &lt;-<span class="st"> "data/bears_fasta.fa"</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3">bears &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(aln_path)</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">tree &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggtree/topics/reexports">read.tree</a></span>(<span class="st">"data/starting_tree.tre"</span>)</a>
<a class="sourceLine" id="cb20-5" data-line-number="5"></a>
<a class="sourceLine" id="cb20-6" data-line-number="6">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/treesiftr/topics/generate_sliding">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">1</span>, <span class="dt">stop_char =</span> <span class="dv">5</span>, <span class="dt">steps =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="kw">print</span>(sample_df)</a></code></pre></div>
<pre><code>  starting_val stop_val step_val
1            1        2        1
2            2        3        1
3            3        4        1
4            4        5        1
5            5        6        1</code></pre>
</div>
<div id="treesiftr-1" class="section level1">
<h1 class="hasAnchor">
<a href="#treesiftr-1" class="anchor"></a>treesiftr</h1>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw">library</span>(phangorn)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw">library</span>(ggtree)</a>
<a class="sourceLine" id="cb22-3" data-line-number="3">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/treesiftr/topics/generate_tree_vis">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                        aln_path,<span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,                                    <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>Final p-score 2 after  0 nni operations 
Final p-score 2 after  0 nni operations 
Final p-score 2 after  0 nni operations 
Final p-score 2 after  1 nni operations 
Final p-score 2 after  1 nni operations </code></pre>
</div>
<div id="treesiftr-2" class="section level1">
<h1 class="hasAnchor">
<a href="#treesiftr-2" class="anchor"></a>treesiftr</h1>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">output_vector <span class="co">#sample output - you will get more than this when you run in your console</span></a></code></pre></div>
<pre><code>[[1]]</code></pre>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-17-1.png" alt="plot of chunk unnamed-chunk-17"></p>
<pre><code>
[[2]]</code></pre>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-17-2.png" alt="plot of chunk unnamed-chunk-17"></p>
<pre><code>
[[3]]</code></pre>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-17-3.png" alt="plot of chunk unnamed-chunk-17"></p>
<pre><code>
[[4]]</code></pre>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-17-4.png" alt="plot of chunk unnamed-chunk-17"></p>
<pre><code>
[[5]]</code></pre>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-17-5.png" alt="plot of chunk unnamed-chunk-17"></p>
<p>??? Do a couple trees on the board, including the pruning algorithm. Then allow them to play.</p>
</div>
<div id="parsimony-many-trees-for-one-character-and-4-taxa" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony-many-trees-for-one-character-and-4-taxa" class="anchor"></a>Parsimony: Many trees for one character and 4 taxa</h1>
<p><img src="img/Ptree_enum.png" alt="Parsimony Trees"></p>
<p>??? This is one character. Imagine many - enumeration is not possible.Also note that several trees have the same “best” tree</p>
</div>
<div id="parsimony-how-do-we-find-the-most-parsimonious-tree" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony-how-do-we-find-the-most-parsimonious-tree" class="anchor"></a>Parsimony: How do we find the most parsimonious tree?</h1>
<ul>
<li>We’re going to take an exercise break and play with PAUP
<ul>
<li><a href="http://phylosolutions.com/paup-test/" class="uri">http://phylosolutions.com/paup-test/</a></li>
<li>Download the command line executeable for your OS</li>
<li>Copy it into this repository</li>
<li>Open a terminal</li>
</ul>
</li>
</ul>
</div>
<div id="paup" class="section level1">
<h1 class="hasAnchor">
<a href="#paup" class="anchor"></a>PAUP</h1>
<pre class="paup"><code>execute data/bears_morphology.nex</code></pre>
<ul>
<li>NOTE: PAUP allows tab-completion</li>
<li>Open the bears_morphology file in a text editor. Now:</li>
</ul>
</div>
<div id="paup-a-couple-important-commands" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-a-couple-important-commands" class="anchor"></a>PAUP: A couple important commands</h1>
<pre class="paup"><code>cstatus
tstatus
showmatrix
showdist
log file="mylogfile"</code></pre>
<ul>
<li>Try each of these - what information do they give you?</li>
</ul>
</div>
<div id="paup-building-a-tree" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-building-a-tree" class="anchor"></a>PAUP: Building a tree</h1>
<pre class="paup"><code>alltrees</code></pre>
<p>What happened here?</p>
</div>
<div id="parsimony-enumeration-is-not-possible-for-more-than-12-taxa" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony-enumeration-is-not-possible-for-more-than-12-taxa" class="anchor"></a>Parsimony: Enumeration is not possible for more than 12 taxa</h1>
<p><img src="img/Ptree_enum.png" alt="Parsimony Trees"></p>
<p>??? This is one character. Imagine many - enumeration is not possible.Also note that several trees have the same “best” tree</p>
</div>
<div id="paup-heuristic-searches" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-heuristic-searches" class="anchor"></a>PAUP: Heuristic Searches</h1>
<p>Heuristic - use of shortcuts to reduce the number of trees we need to search</p>
<pre class="paup"><code>hsearch</code></pre>
<ul>
<li>What is the name of the heuristic that was used?</li>
<li>How was the initial tree discovered?</li>
<li>How many trees were searched?</li>
<li>How many “best” trees were there, and what is their score?</li>
</ul>
</div>
<div id="paup-heuristic-searches-1" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-heuristic-searches-1" class="anchor"></a>PAUP: Heuristic Searches</h1>
<p>Heuristic - use of shortcuts to reduce the number of trees we need to search</p>
<pre class="paup"><code>hsearch swap = nni</code></pre>
<ul>
<li>How many trees were examined with this algorithm? Why is this number so much smaller?</li>
<li>How many “best” trees were found, and what is their score?</li>
</ul>
</div>
<div id="paup-heuristic-searches-2" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-heuristic-searches-2" class="anchor"></a>PAUP: Heuristic Searches</h1>
<p>Heuristic - use of shortcuts to reduce the number of trees we need to search</p>
<pre class="paup"><code>hsearch swap = spr</code></pre>
<ul>
<li>How many trees were examined with this algorithm?</li>
<li>How many “best” trees were found, and what is their score?</li>
<li>When would we expect searching algorithm to matter strongly?</li>
</ul>
</div>
<div id="paup-exporting-parsimony-trees" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-exporting-parsimony-trees" class="anchor"></a>PAUP: Exporting parsimony trees</h1>
<pre class="paup"><code>savetrees from=1 to=1 file=results/tree1.tre;
savetrees from=2 to=2 file=results/tree2.tre;
savetrees from=3 to=3 file=results/tree3.tre;
</code></pre>
</div>
<div id="paup-reading-in-parsimony-trees" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-reading-in-parsimony-trees" class="anchor"></a>PAUP: Reading in parsimony trees</h1>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="kw">library</span>(ape)</a>
<a class="sourceLine" id="cb37-2" data-line-number="2">tree1 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.nexus">read.nexus</a></span>(<span class="st">"results/tree1.tre"</span>)</a>
<a class="sourceLine" id="cb37-3" data-line-number="3">tree2 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.nexus">read.nexus</a></span>(<span class="st">"results/tree2.tre"</span>)</a>
<a class="sourceLine" id="cb37-4" data-line-number="4">tree3 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.nexus">read.nexus</a></span>(<span class="st">"results/tree3.tre"</span>)</a></code></pre></div>
</div>
<div id="paup-what-do-we-do-with-multiple-best-trees" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-what-do-we-do-with-multiple-best-trees" class="anchor"></a>PAUP: What do we do with multiple “best” trees?</h1>
<p><img src="img/Ptree_enum.png" alt="Parsimony Trees"></p>
<p>??? This is one character. Imagine many - enumeration is not possible.Also note that several trees have the same “best” tree</p>
</div>
<div id="paup-what-do-we-do-with-multiple-best-trees-1" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-what-do-we-do-with-multiple-best-trees-1" class="anchor"></a>PAUP: What do we do with multiple “best” trees?</h1>
<ul>
<li>Typically: Build a consensus tree</li>
</ul>
<pre class="paup"><code>  contree all / treefile=Results/contree.tre;
  help contree</code></pre>
<ul>
<li>Please look at:
<ul>
<li>Semistrict</li>
<li>Majority rule</li>
<li>Adams</li>
</ul>
</li>
</ul>
</div>
<div id="how-do-we-assess-confidence-in-a-tree" class="section level1">
<h1 class="hasAnchor">
<a href="#how-do-we-assess-confidence-in-a-tree" class="anchor"></a>How do we assess confidence in a tree?</h1>
<ul>
<li>The bootstrap</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">char_mat &lt;-<span class="st"> </span>phangorn<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/phangorn/topics/phyDat">phyDat</a></span>(char_data, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="st">"?"</span>), <span class="dt">type =</span> <span class="st">"USER"</span>)</a>
<a class="sourceLine" id="cb39-2" data-line-number="2">samples &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">62</span>, <span class="dv">62</span>, <span class="dt">replace =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">new_mat &lt;-<span class="st"> </span><span class="kw">subset</span>(char_mat, <span class="dt">select=</span>samples)</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">samples</a></code></pre></div>
<pre><code> [1]  6  8 22 13 11 45 16  6 43 33 43 32 47 60 59 25 14 33  2 18 22 54 25
[24] 12 21 20 56 43  3 12 10 34 57 25  9 18 22 20 12 53  1 12 20 44 40 22
[47] 18 12  6 21  3  4 54 55 25 59 48 15 38 25  6 42</code></pre>
</div>
<div id="how-do-we-assess-confidence-in-a-tree-1" class="section level1">
<h1 class="hasAnchor">
<a href="#how-do-we-assess-confidence-in-a-tree-1" class="anchor"></a>How do we assess confidence in a tree?</h1>
<ul>
<li>The bootstrap samples characters <em>with</em> replacement</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">char_mat &lt;-<span class="st"> </span>phangorn<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/phangorn/topics/phyDat">phyDat</a></span>(char_data, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="st">"?"</span>), <span class="dt">type =</span> <span class="st">"USER"</span>)</a>
<a class="sourceLine" id="cb41-2" data-line-number="2">list_of_mats &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb41-3" data-line-number="3"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>){</a>
<a class="sourceLine" id="cb41-4" data-line-number="4">samples &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">62</span>, <span class="dv">62</span>, <span class="dt">replace =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb41-5" data-line-number="5">new_mat &lt;-<span class="st"> </span><span class="kw">subset</span>(char_mat, <span class="dt">select=</span>samples)</a>
<a class="sourceLine" id="cb41-6" data-line-number="6">list_of_mats[[i]] &lt;-<span class="st"> </span>new_mat</a>
<a class="sourceLine" id="cb41-7" data-line-number="7">}</a></code></pre></div>
<p>Generates a tree from each matrix</p>
</div>
<div id="the-bootstrap" class="section level1">
<h1 class="hasAnchor">
<a href="#the-bootstrap" class="anchor"></a>The bootstrap</h1>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw">plot</span>(tree1)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-29-1.png" alt="plot of chunk unnamed-chunk-29"></p>
</div>
<div id="paup-the-bootstrap" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-the-bootstrap" class="anchor"></a>PAUP: The bootstrap</h1>
<pre class="paup"><code>bootstrap nreps=100 search=heuristic /addseq=random;
savetrees from=1 to=1 savebootp=nodelabels file=results/bootstrap.tre;</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">b_tre &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.nexus">read.nexus</a></span>(<span class="st">"results/bootstrap.tre"</span>)</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"><span class="kw">plot</span>(b_tre)</a>
<a class="sourceLine" id="cb44-3" data-line-number="3"><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/nodelabels">nodelabels</a></span>(b_tre<span class="op">$</span>nodelabels)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-31-1.png" alt="plot of chunk unnamed-chunk-31"></p>
</div>
<div id="paup-the-bootstrap-1" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-the-bootstrap-1" class="anchor"></a>PAUP: The bootstrap</h1>
<pre class="paup"><code>bootstrap nreps=100 search=heuristic /addseq=random;
savetrees from=1 to=1 savebootp=nodelabels file=results/bootstrap.tre;</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">b_tre &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.nexus">read.nexus</a></span>(<span class="st">"results/bootstrap.tre"</span>)</a>
<a class="sourceLine" id="cb46-2" data-line-number="2"><span class="kw">plot</span>(b_tre)</a>
<a class="sourceLine" id="cb46-3" data-line-number="3"><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/nodelabels">nodelabels</a></span>(b_tre<span class="op">$</span>node.label)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-33-1.png" alt="plot of chunk unnamed-chunk-33"> “b_tre$nodelabels” what is this construction?</p>
</div>
<div id="paup-the-jacknife" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-the-jacknife" class="anchor"></a>PAUP: The jacknife</h1>
<p>Samples without replacement</p>
<pre class="paup"><code>jackknife nreps=100 search=heuristic /addseq=random;
savetrees from=1 to=1 savebootp=nodelabels file=results/jackknife.tre;</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">j_tre &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.nexus">read.nexus</a></span>(<span class="st">"results/jackknife.tre"</span>)</a>
<a class="sourceLine" id="cb48-2" data-line-number="2"><span class="kw">plot</span>(j_tre)</a>
<a class="sourceLine" id="cb48-3" data-line-number="3"><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/nodelabels">nodelabels</a></span>(j_tre<span class="op">$</span>node.label)</a></code></pre></div>
<p><img src="TreeThinking.Rmd-figure/unnamed-chunk-35-1.png" alt="plot of chunk unnamed-chunk-35"></p>
</div>
<div id="parsimony-some-of-these-trees-imply-homoplasy" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony-some-of-these-trees-imply-homoplasy" class="anchor"></a>Parsimony: Some of these trees imply homoplasy</h1>
<p><img src="img/Ptree_enum.png" alt="Parsimony Trees"></p>
<p>Is an answer that implies homoplasy ever the right one?</p>
</div>
<div id="parsimony-different-step-matrices" class="section level1">
<h1 class="hasAnchor">
<a href="#parsimony-different-step-matrices" class="anchor"></a>Parsimony: Different Step Matrices</h1>
<ul>
<li>Can make different character interchanges weigh more or less than others</li>
<li>For example, for character 1, it might be more likely to go from 0 to 1 than the reverse</li>
<li>Let’s try this in PAUP</li>
</ul>
</div>
<div id="paup-different-step-matrices" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-different-step-matrices" class="anchor"></a>PAUP: Different Step Matrices</h1>
<ul>
<li>Open the bears_morphology file in a test editor. Scroll all the way to the bottom and paste in:</li>
</ul>
<pre class="paup"><code>begin assumptions;
 usertype my_ctype stepmatrix=2
     0 1
 [0] . 1
 [1] 0 .

 ;
end;</code></pre>
</div>
<div id="paup-different-step-matrices-1" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-different-step-matrices-1" class="anchor"></a>PAUP: Different Step Matrices</h1>
<ul>
<li>In the PAUP executeable:</li>
</ul>
<pre class="paup"><code>ctype my_ctype:1
hsearch</code></pre>
</div>
<div id="paup-different-step-matrices-2" class="section level1">
<h1 class="hasAnchor">
<a href="#paup-different-step-matrices-2" class="anchor"></a>PAUP: Different Step Matrices</h1>
<ul>
<li><p>For those of you who are more familiar with likelihood and Bayesian analyses, what does this sound like?</p></li>
<li><p>We’ll pick up this afternoon with Bayesian models</p></li>
</ul>
</div>


  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

